{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LRRK2 p.A419V - Haplotype analysis "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Project: Multiancestry LRRK2 p.A419V analysis\n",
    "- Version: Python/3.10.12\n",
    "- Last Updated: 05-MAY-2025"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Description\n",
    "\n",
    "**1. Description**\n",
    "- Getting started\n",
    "- Load python libraries\n",
    "- Define function\n",
    "- Setting up path\n",
    "- Install R and its packages\n",
    "\n",
    "**2. Haplotype block size comparison**\n",
    "\n",
    "**3. Haplotype association study**\n",
    "\n",
    "**4. R2 calculation**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_1062/1504764740.py:21: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Use the os package to interact with the environment\n",
    "import os\n",
    "\n",
    "# Bring in Pandas for Dataframe functionality\n",
    "import pandas as pd\n",
    "\n",
    "# Numpy for basics\n",
    "import numpy as np\n",
    "\n",
    "# Use StringIO for working with file contents\n",
    "from io import StringIO\n",
    "\n",
    "# Enable IPython to display matplotlib graphs\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# Enable interaction with the FireCloud API\n",
    "from firecloud import api as fapi\n",
    "\n",
    "# Import the iPython HTML rendering for displaying links to Google Cloud Console\n",
    "from IPython.core.display import display, HTML\n",
    "\n",
    "# Import urllib modules for building URLs to Google Cloud Console\n",
    "import urllib.parse\n",
    "\n",
    "# BigQuery for querying data\n",
    "from google.cloud import bigquery\n",
    "\n",
    "#Import Sys\n",
    "import sys as sys"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)\n",
    "\n",
    "# Utility routine for printing a query before executing it\n",
    "def bq_query(query):\n",
    "    print(f'Executing: {query}', file=sys.stderr)\n",
    "    return pd.read_gbq(query, project_id=BILLING_PROJECT_ID, dialect='standard')\n",
    "\n",
    "# Utility routine for display a message and a link\n",
    "def display_html_link(description, link_text, url):\n",
    "    html = f'''\n",
    "    <p>\n",
    "    </p>\n",
    "    <p>\n",
    "    {description}\n",
    "    <a target=_blank href=\"{url}\">{link_text}</a>.\n",
    "    </p>\n",
    "    '''\n",
    "\n",
    "    display(HTML(html))\n",
    "\n",
    "# Utility routines for reading files from Google Cloud Storage\n",
    "def gcs_read_file(path):\n",
    "    \"\"\"Return the contents of a file in GCS\"\"\"\n",
    "    contents = !gsutil -u {BILLING_PROJECT_ID} cat {path}\n",
    "    return '\\n'.join(contents)\n",
    "    \n",
    "def gcs_read_csv(path, sep=None):\n",
    "    \"\"\"Return a DataFrame from the contents of a delimited file in GCS\"\"\"\n",
    "    return pd.read_csv(StringIO(gcs_read_file(path)), sep=sep, engine='python')\n",
    "\n",
    "# Utility routine for displaying a message and link to Cloud Console\n",
    "def link_to_cloud_console_gcs(description, link_text, gcs_path):\n",
    "    url = '{}?{}'.format(\n",
    "        os.path.join('https://console.cloud.google.com/storage/browser',\n",
    "                     gcs_path.replace(\"gs://\",\"\")),\n",
    "        urllib.parse.urlencode({'userProject': BILLING_PROJECT_ID}))\n",
    "\n",
    "    display_html_link(description, link_text, url)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Setting up path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GP2 v5.0\n",
      "Path to GP2 v2.0 Clinical Data: gs://gp2tier2/release9_18122024/clinical_data\n",
      "Path to GP2 v2.0 Raw Genotype Data: gs://gp2tier2/release9_18122024/raw_genotypes\n",
      "Path to GP2 v2.0 Imputed Genotype Data: gs://gp2tier2/release9_18122024/imputed_genotypes\n"
     ]
    }
   ],
   "source": [
    "# Set up billing project and data path variables\n",
    "BILLING_PROJECT_ID = os.environ['GOOGLE_PROJECT']\n",
    "WORKSPACE_NAMESPACE = os.environ['WORKSPACE_NAMESPACE']\n",
    "WORKSPACE_NAME = os.environ['WORKSPACE_NAME']\n",
    "WORKSPACE_BUCKET = os.environ['WORKSPACE_BUCKET']\n",
    "\n",
    "WORKSPACE_ATTRIBUTES = fapi.get_workspace(WORKSPACE_NAMESPACE, WORKSPACE_NAME).json().get('workspace',{}).get('attributes',{})\n",
    "\n",
    "## GP2 v5.0 gs://gp2tier2/release9_18122024/\n",
    "GP2_TIER1 = 'gs://gp2tier1/release9_18122024'\n",
    "GP2_RELEASE_PATH = 'gs://gp2tier2/release9_18122024'\n",
    "GP2_CLINICAL_RELEASE_PATH = f'{GP2_RELEASE_PATH}/clinical_data'\n",
    "GP2_META_RELEASE_PATH = f'{GP2_RELEASE_PATH}/meta_data'\n",
    "GP2_SUMSTAT_RELEASE_PATH = f'{GP2_RELEASE_PATH}/summary_statistics'\n",
    "\n",
    "GP2_RAW_GENO_PATH = f'{GP2_RELEASE_PATH}/raw_genotypes'\n",
    "GP2_IMPUTED_GENO_PATH = f'{GP2_RELEASE_PATH}/imputed_genotypes'\n",
    "print('GP2 v5.0')\n",
    "print(f'Path to GP2 v2.0 Clinical Data: {GP2_CLINICAL_RELEASE_PATH}')\n",
    "print(f'Path to GP2 v2.0 Raw Genotype Data: {GP2_RAW_GENO_PATH}')\n",
    "print(f'Path to GP2 v2.0 Imputed Genotype Data: {GP2_IMPUTED_GENO_PATH}')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R and its packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir -p /home/jupyter/A419V_release9/R_packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Install and import rpy2\n",
    "!pip install rpy2\n",
    "import rpy2.rinterface"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The rpy2.ipython extension is already loaded. To reload it, use:\n",
      "  %reload_ext rpy2.ipython\n"
     ]
    }
   ],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  library ‘/usr/lib/R/site-library’ contains no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "BILLING_PROJECT_ID  <- Sys.getenv('GOOGLE_PROJECT')\n",
    "WORKSPACE_NAMESPACE <- Sys.getenv('WORKSPACE_NAMESPACE')\n",
    "WORKSPACE_NAME      <- Sys.getenv('WORKSPACE_NAME')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "* installing *source* package ‘dplyr’ ...\n",
      "** package ‘dplyr’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** libs\n",
      "using C++ compiler: ‘g++ (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c chop.cpp -o chop.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c filter.cpp -o filter.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c funs.cpp -o funs.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c group_by.cpp -o group_by.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c group_data.cpp -o group_data.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c imports.cpp -o imports.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c init.cpp -o init.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c mask.cpp -o mask.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c mutate.cpp -o mutate.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c reconstruct.cpp -o reconstruct.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c slice.cpp -o slice.o\n",
      "g++ -std=gnu++17 -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2   -c summarise.cpp -o summarise.o\n",
      "g++ -std=gnu++17 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o dplyr.so chop.o filter.o funs.o group_by.o group_data.o imports.o init.o mask.o mutate.o reconstruct.o slice.o summarise.o -L/usr/lib/R/lib -lR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "installing to /home/jupyter/A419V_release9/R_packages/00LOCK-dplyr/00new/dplyr/libs\n",
      "** R\n",
      "** data\n",
      "*** moving datasets to lazyload DB\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "*** copying figures\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** checking absolute paths in shared objects and dynamic libraries\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (dplyr)\n",
      "* installing *source* package ‘data.table’ ...\n",
      "** package ‘data.table’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc 9.4.0\n",
      "zlib 1.2.11 is available ok\n",
      "* checking if R installation supports OpenMP without any extra hints... yes\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c assign.c -o assign.o\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "** libs\n",
      "using C compiler: ‘gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c between.c -o between.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c bmerge.c -o bmerge.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c chmatch.c -o chmatch.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c cj.c -o cj.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c coalesce.c -o coalesce.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dogroups.c -o dogroups.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fastmean.c -o fastmean.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fcast.c -o fcast.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fifelse.c -o fifelse.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fmelt.c -o fmelt.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c forder.c -o forder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frank.c -o frank.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fread.c -o fread.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c freadR.c -o freadR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c froll.c -o froll.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frollR.c -o frollR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frolladaptive.c -o frolladaptive.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fsort.c -o fsort.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwrite.c -o fwrite.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwriteR.c -o fwriteR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c gsumm.c -o gsumm.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c idatetime.c -o idatetime.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ijoin.c -o ijoin.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c init.c -o init.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c inrange.c -o inrange.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nafill.c -o nafill.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c negate.c -o negate.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nqrecreateindices.c -o nqrecreateindices.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c openmp-utils.c -o openmp-utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c programming.c -o programming.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c quickselect.c -o quickselect.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rbindlist.c -o rbindlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c reorder.c -o reorder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c shift.c -o shift.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c snprintf.c -o snprintf.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c subset.c -o subset.o\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c transpose.c -o transpose.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c types.c -o types.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c uniqlist.c -o uniqlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c utils.c -o utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c vecseq.c -o vecseq.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c wrappers.c -o wrappers.o\n",
      "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o data.table.so assign.o between.o bmerge.o chmatch.o cj.o coalesce.o dogroups.o fastmean.o fcast.o fifelse.o fmelt.o forder.o frank.o fread.o freadR.o froll.o frollR.o frolladaptive.o fsort.o fwrite.o fwriteR.o gsumm.o idatetime.o ijoin.o init.o inrange.o nafill.o negate.o nqrecreateindices.o openmp-utils.o programming.o quickselect.o rbindlist.o reorder.o shift.o snprintf.o subset.o transpose.o types.o uniqlist.o utils.o vecseq.o wrappers.o -fopenmp -L/usr/lib/x86_64-linux-gnu -lz -L/usr/lib/R/lib -lR\n",
      "PKG_CFLAGS = -fopenmp\n",
      "PKG_LIBS = -fopenmp -L/usr/lib/x86_64-linux-gnu -lz\n",
      "if [ \"data.table.so\" != \"data_table.so\" ]; then mv data.table.so data_table.so; fi\n",
      "if [ \"\" != \"Windows_NT\" ] && [ `uname -s` = 'Darwin' ]; then install_name_tool -id data_table.so data_table.so; fi\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "installing to /home/jupyter/A419V_release9/R_packages/00LOCK-data.table/00new/data.table/libs\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** checking absolute paths in shared objects and dynamic libraries\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (data.table)\n",
      "* installing *source* package ‘arsenal’ ...\n",
      "** package ‘arsenal’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** R\n",
      "** data\n",
      "*** moving datasets to lazyload DB\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "*** copying figures\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (arsenal)\n",
      "* installing *source* package ‘haplo.stats’ ...\n",
      "** package ‘haplo.stats’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** libs\n",
      "using C compiler: ‘gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c groupsum.c -o groupsum.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c haplo.stats_init.c -o haplo.stats_init.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c haplo_em_pin.c -o haplo_em_pin.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c louis_info.c -o louis_info.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG       -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c seqhap.c -o seqhap.o\n",
      "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o haplo.stats.so groupsum.o haplo.stats_init.o haplo_em_pin.o louis_info.o seqhap.o -L/usr/lib/R/lib -lR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "installing to /home/jupyter/A419V_release9/R_packages/00LOCK-haplo.stats/00new/haplo.stats/libs\n",
      "** R\n",
      "** data\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** checking absolute paths in shared objects and dynamic libraries\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (haplo.stats)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "trying URL 'https://cloud.r-project.org/src/contrib/dplyr_1.1.4.tar.gz'\n",
       "Content type 'application/x-gzip' length 1207521 bytes (1.2 MB)\n",
       "==================================================\n",
       "downloaded 1.2 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/Rtmp8Akrfw/downloaded_packages’\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/data.table_1.17.0.tar.gz'\n",
       "Content type 'application/x-gzip' length 5833671 bytes (5.6 MB)\n",
       "==================================================\n",
       "downloaded 5.6 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/Rtmp8Akrfw/downloaded_packages’\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/arsenal_3.6.3.tar.gz'\n",
       "Content type 'application/x-gzip' length 672939 bytes (657 KB)\n",
       "==================================================\n",
       "downloaded 657 KB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/Rtmp8Akrfw/downloaded_packages’\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/haplo.stats_1.9.7.tar.gz'\n",
       "Content type 'application/x-gzip' length 2766759 bytes (2.6 MB)\n",
       "==================================================\n",
       "downloaded 2.6 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/Rtmp8Akrfw/downloaded_packages’\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "pack <- \"/home/jupyter/A419V_release9/R_packages\"\n",
    "\n",
    "install.packages(\"dplyr\", lib = pack)\n",
    "install.packages(\"data.table\", lib = pack)\n",
    "install.packages(\"arsenal\", lib = pack)\n",
    "install.packages(\"haplo.stats\", lib = pack)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "pack <- \"/home/jupyter/A419V_release9/R_packages\"\n",
    "\n",
    "suppressPackageStartupMessages(library(dplyr, lib.loc = pack))\n",
    "suppressPackageStartupMessages(library(data.table, lib.loc = pack))\n",
    "suppressPackageStartupMessages(library(arsenal, lib.loc = pack))\n",
    "suppressPackageStartupMessages(library(haplo.stats, lib.loc = pack))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AAC/AAC_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile AAC/AAC_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out AAC/AAC_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "448 out of 1896835 variants loaded from .bim file.\n",
      "1207 people (498 males, 709 females) loaded from .fam.\n",
      "1165 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 1207 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998097.\n",
      "--missing: Sample missing data report written to\n",
      "AAC/AAC_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to AAC/AAC_release9_remove_related_lrrk2.lmiss.\n",
      "448 variants and 1207 people pass filters and QC.\n",
      "Among remaining phenotypes, 338 are cases and 827 are controls.  (42 phenotypes\n",
      "are missing.)\n",
      "--make-bed to AAC/AAC_release9_remove_related_lrrk2.bed +\n",
      "AAC/AAC_release9_remove_related_lrrk2.bim +\n",
      "AAC/AAC_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR/AFR_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile AFR/AFR_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out AFR/AFR_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "426 out of 1788859 variants loaded from .bim file.\n",
      "2680 people (1482 males, 1198 females) loaded from .fam.\n",
      "2650 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 2680 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998112.\n",
      "--missing: Sample missing data report written to\n",
      "AFR/AFR_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to AFR/AFR_release9_remove_related_lrrk2.lmiss.\n",
      "426 variants and 2680 people pass filters and QC.\n",
      "Among remaining phenotypes, 983 are cases and 1667 are controls.  (30\n",
      "phenotypes are missing.)\n",
      "--make-bed to AFR/AFR_release9_remove_related_lrrk2.bed +\n",
      "AFR/AFR_release9_remove_related_lrrk2.bim +\n",
      "AFR/AFR_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AJ/AJ_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile AJ/AJ_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out AJ/AJ_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "447 out of 1875579 variants loaded from .bim file.\n",
      "3081 people (1934 males, 1147 females) loaded from .fam.\n",
      "2533 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 3081 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998107.\n",
      "--missing: Sample missing data report written to\n",
      "AJ/AJ_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to AJ/AJ_release9_remove_related_lrrk2.lmiss.\n",
      "447 variants and 3081 people pass filters and QC.\n",
      "Among remaining phenotypes, 1709 are cases and 824 are controls.  (548\n",
      "phenotypes are missing.)\n",
      "--make-bed to AJ/AJ_release9_remove_related_lrrk2.bed +\n",
      "AJ/AJ_release9_remove_related_lrrk2.bim +\n",
      "AJ/AJ_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AMR/AMR_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile AMR/AMR_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out AMR/AMR_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "441 out of 1843508 variants loaded from .bim file.\n",
      "3493 people (1656 males, 1837 females) loaded from .fam.\n",
      "3433 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 3493 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998813.\n",
      "--missing: Sample missing data report written to\n",
      "AMR/AMR_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to AMR/AMR_release9_remove_related_lrrk2.lmiss.\n",
      "441 variants and 3493 people pass filters and QC.\n",
      "Among remaining phenotypes, 2005 are cases and 1428 are controls.  (60\n",
      "phenotypes are missing.)\n",
      "--make-bed to AMR/AMR_release9_remove_related_lrrk2.bed +\n",
      "AMR/AMR_release9_remove_related_lrrk2.bim +\n",
      "AMR/AMR_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out CAH/CAH_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "446 out of 1904683 variants loaded from .bim file.\n",
      "982 people (514 males, 468 females) loaded from .fam.\n",
      "954 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 982 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998123.\n",
      "--missing: Sample missing data report written to\n",
      "CAH/CAH_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to CAH/CAH_release9_remove_related_lrrk2.lmiss.\n",
      "446 variants and 982 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 310 are controls.  (28 phenotypes\n",
      "are missing.)\n",
      "--make-bed to CAH/CAH_release9_remove_related_lrrk2.bed +\n",
      "CAH/CAH_release9_remove_related_lrrk2.bim +\n",
      "CAH/CAH_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Logging to CAS/CAS_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out CAS/CAS_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "444 out of 1902516 variants loaded from .bim file.\n",
      "1006 people (461 males, 545 females) loaded from .fam.\n",
      "990 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 1006 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998117.\n",
      "--missing: Sample missing data report written to\n",
      "CAS/CAS_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to CAS/CAS_release9_remove_related_lrrk2.lmiss.\n",
      "444 variants and 1006 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 329 are controls.  (16 phenotypes\n",
      "are missing.)\n",
      "--make-bed to CAS/CAS_release9_remove_related_lrrk2.bed +\n",
      "CAS/CAS_release9_remove_related_lrrk2.bim +\n",
      "CAS/CAS_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out EAS/EAS_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "427 out of 1730431 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 5646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998657.\n",
      "--missing: Sample missing data report written to\n",
      "EAS/EAS_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to EAS/EAS_release9_remove_related_lrrk2.lmiss.\n",
      "427 variants and 5646 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 2379 are controls.  (75\n",
      "phenotypes are missing.)\n",
      "--make-bed to EAS/EAS_release9_remove_related_lrrk2.bed +\n",
      "EAS/EAS_release9_remove_related_lrrk2.bim +\n",
      "EAS/EAS_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out EUR/EUR_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "409 out of 1668646 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 34703 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998566.\n",
      "--missing: Sample missing data report written to\n",
      "EUR/EUR_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to EUR/EUR_release9_remove_related_lrrk2.lmiss.\n",
      "409 variants and 34703 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 5492 are controls.  (13879\n",
      "phenotypes are missing.)\n",
      "--make-bed to EUR/EUR_release9_remove_related_lrrk2.bed +\n",
      "EUR/EUR_release9_remove_related_lrrk2.bim +\n",
      "EUR/EUR_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to FIN/FIN_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile FIN/FIN_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out FIN/FIN_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "451 out of 1883041 variants loaded from .bim file.\n",
      "109 people (43 males, 66 females) loaded from .fam.\n",
      "90 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 109 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998352.\n",
      "--missing: Sample missing data report written to\n",
      "FIN/FIN_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to FIN/FIN_release9_remove_related_lrrk2.lmiss.\n",
      "451 variants and 109 people pass filters and QC.\n",
      "Among remaining phenotypes, 86 are cases and 4 are controls.  (19 phenotypes\n",
      "are missing.)\n",
      "--make-bed to FIN/FIN_release9_remove_related_lrrk2.bed +\n",
      "FIN/FIN_release9_remove_related_lrrk2.bim +\n",
      "FIN/FIN_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MDE/MDE_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile MDE/MDE_release9_remove_related_updated\n",
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out MDE/MDE_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "448 out of 1888776 variants loaded from .bim file.\n",
      "712 people (438 males, 274 females) loaded from .fam.\n",
      "678 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 712 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998486.\n",
      "--missing: Sample missing data report written to\n",
      "MDE/MDE_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to MDE/MDE_release9_remove_related_lrrk2.lmiss.\n",
      "448 variants and 712 people pass filters and QC.\n",
      "Among remaining phenotypes, 481 are cases and 197 are controls.  (34 phenotypes\n",
      "are missing.)\n",
      "--make-bed to MDE/MDE_release9_remove_related_lrrk2.bed +\n",
      "MDE/MDE_release9_remove_related_lrrk2.bim +\n",
      "MDE/MDE_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SAS/SAS_release9_remove_related_lrrk2.log.\n",
      "Options in effect:\n",
      "  --bfile SAS/SAS_release9_remove_related_updated\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  --chr 12\n",
      "  --from-bp 40196744\n",
      "  --make-bed\n",
      "  --missing\n",
      "  --out SAS/SAS_release9_remove_related_lrrk2\n",
      "  --to-bp 40369285\n",
      "\n",
      "3672 MB RAM detected; reserving 1836 MB for main workspace.\n",
      "449 out of 1919616 variants loaded from .bim file.\n",
      "580 people (375 males, 205 females) loaded from .fam.\n",
      "553 phenotype values loaded from .fam.\n",
      "Using 1 thread.\n",
      "Before main variant filters, 580 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998629.\n",
      "--missing: Sample missing data report written to\n",
      "SAS/SAS_release9_remove_related_lrrk2.imiss, and variant-based missing data\n",
      "report written to SAS/SAS_release9_remove_related_lrrk2.lmiss.\n",
      "449 variants and 580 people pass filters and QC.\n",
      "Among remaining phenotypes, 354 are cases and 199 are controls.  (27 phenotypes\n",
      "are missing.)\n",
      "--make-bed to SAS/SAS_release9_remove_related_lrrk2.bed +\n",
      "SAS/SAS_release9_remove_related_lrrk2.bim +\n",
      "SAS/SAS_release9_remove_related_lrrk2.fam ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'FIN' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_updated \\\n",
    "    --chr 12 \\\n",
    "    --from-bp 40196744 \\\n",
    "    --to-bp 40369285 \\\n",
    "    --missing \\\n",
    "    --make-bed \\\n",
    "    --out ${label}/${label}_release9_remove_related_lrrk2\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "ancestry_labels=['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS']\n",
    "\n",
    "for anc in ancestry_labels:\n",
    "    \n",
    "    bim = pd.read_csv(f\"/home/jupyter/A419V_release9/{anc}/{anc}_release9_remove_related_lrrk2.bim\", \n",
    "                      sep = \"\\t\", names = [\"CHR\", \"RSID\", \"POS\", \"BP\", \"A1\", \"A2\"])\n",
    "    bim[\"CHR\"] = bim[\"CHR\"].astype(str)\n",
    "    bim[\"BP\"] = bim[\"BP\"].astype(str)\n",
    "    bim[\"RSID\"] = bim[\"CHR\"].str.cat(bim[\"BP\"], sep = \"_\")\n",
    "    bim[\"RSID\"] = bim[\"RSID\"].str.cat(bim[\"A2\"], sep = \"_\")\n",
    "    bim[\"RSID\"] = bim[\"RSID\"].str.cat(bim[\"A1\"], sep = \"_\")\n",
    "    \n",
    "    bim.to_csv(f\"/home/jupyter/A419V_release9/{anc}/{anc}_release9_remove_related_lrrk2.bim\", \n",
    "               sep = \"\\t\", index = False, header = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AAC/AAC_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile AAC/AAC_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out AAC/AAC_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2016 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "1207 samples (709 females, 498 males; 1207 founders) loaded from\n",
      "AAC/AAC_release9_remove_related_lrrk2.fam.\n",
      "448 variants loaded from AAC/AAC_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (338 cases, 827 controls).\n",
      "--rm-dup: 65 duplicated IDs, 83 variants removed.\n",
      "Writing AAC/AAC_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing AAC/AAC_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing AAC/AAC_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AFR/AFR_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile AFR/AFR_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out AFR/AFR_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2016 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "2680 samples (1198 females, 1482 males; 2680 founders) loaded from\n",
      "AFR/AFR_release9_remove_related_lrrk2.fam.\n",
      "426 variants loaded from AFR/AFR_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (983 cases, 1667 controls).\n",
      "--rm-dup: 63 duplicated IDs, 78 variants removed.\n",
      "Writing AFR/AFR_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing AFR/AFR_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing AFR/AFR_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AJ/AJ_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile AJ/AJ_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out AJ/AJ_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2015 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "3081 samples (1147 females, 1934 males; 3081 founders) loaded from\n",
      "AJ/AJ_release9_remove_related_lrrk2.fam.\n",
      "447 variants loaded from AJ/AJ_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (1709 cases, 824 controls).\n",
      "--rm-dup: 64 duplicated IDs, 82 variants removed.\n",
      "Writing AJ/AJ_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing AJ/AJ_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing AJ/AJ_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AMR/AMR_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile AMR/AMR_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out AMR/AMR_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2015 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "3493 samples (1837 females, 1656 males; 3493 founders) loaded from\n",
      "AMR/AMR_release9_remove_related_lrrk2.fam.\n",
      "441 variants loaded from AMR/AMR_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (2005 cases, 1428 controls).\n",
      "--rm-dup: 64 duplicated IDs, 81 variants removed.\n",
      "Writing AMR/AMR_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing AMR/AMR_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing AMR/AMR_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out CAH/CAH_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2015 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "982 samples (468 females, 514 males; 982 founders) loaded from\n",
      "CAH/CAH_release9_remove_related_lrrk2.fam.\n",
      "446 variants loaded from CAH/CAH_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (644 cases, 310 controls).\n",
      "--rm-dup: 64 duplicated IDs, 82 variants removed.\n",
      "Writing CAH/CAH_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing CAH/CAH_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing CAH/CAH_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out CAS/CAS_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2015 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "1006 samples (545 females, 461 males; 1006 founders) loaded from\n",
      "CAS/CAS_release9_remove_related_lrrk2.fam.\n",
      "444 variants loaded from CAS/CAS_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (661 cases, 329 controls).\n",
      "--rm-dup: 64 duplicated IDs, 82 variants removed.\n",
      "Writing CAS/CAS_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing CAS/CAS_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing CAS/CAS_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2015 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "5646 samples (2151 females, 3495 males; 5646 founders) loaded from\n",
      "EAS/EAS_release9_remove_related_lrrk2.fam.\n",
      "427 variants loaded from EAS/EAS_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (3192 cases, 2379 controls).\n",
      "--rm-dup: 63 duplicated IDs, 77 variants removed.\n",
      "Writing EAS/EAS_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing EAS/EAS_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing EAS/EAS_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out EUR/EUR_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2014 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "34703 samples (15464 females, 19239 males; 34703 founders) loaded from\n",
      "EUR/EUR_release9_remove_related_lrrk2.fam.\n",
      "409 variants loaded from EUR/EUR_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (15332 cases, 5492 controls).\n",
      "--rm-dup: 60 duplicated IDs, 77 variants removed.\n",
      "Writing EUR/EUR_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing EUR/EUR_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing EUR/EUR_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to FIN/FIN_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile FIN/FIN_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out FIN/FIN_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2006 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "109 samples (66 females, 43 males; 109 founders) loaded from\n",
      "FIN/FIN_release9_remove_related_lrrk2.fam.\n",
      "451 variants loaded from FIN/FIN_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (86 cases, 4 controls).\n",
      "--rm-dup: 66 duplicated IDs, 84 variants removed.\n",
      "Writing FIN/FIN_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing FIN/FIN_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing FIN/FIN_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to MDE/MDE_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile MDE/MDE_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out MDE/MDE_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2007 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "712 samples (274 females, 438 males; 712 founders) loaded from\n",
      "MDE/MDE_release9_remove_related_lrrk2.fam.\n",
      "448 variants loaded from MDE/MDE_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (481 cases, 197 controls).\n",
      "--rm-dup: 66 duplicated IDs, 84 variants removed.\n",
      "Writing MDE/MDE_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing MDE/MDE_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing MDE/MDE_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to SAS/SAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile SAS/SAS_release9_remove_related_lrrk2\n",
      "  --make-bed\n",
      "  --out SAS/SAS_release9_remove_related_lrrk2_nodup\n",
      "  --rm-dup force-first\n",
      "\n",
      "Start time: Sun Apr  6 00:05:04 2025\n",
      "3672 MiB RAM detected, ~2007 available; reserving 1836 MiB for main workspace.\n",
      "Using 1 compute thread.\n",
      "580 samples (205 females, 375 males; 580 founders) loaded from\n",
      "SAS/SAS_release9_remove_related_lrrk2.fam.\n",
      "449 variants loaded from SAS/SAS_release9_remove_related_lrrk2.bim.\n",
      "1 binary phenotype loaded (354 cases, 199 controls).\n",
      "--rm-dup: 66 duplicated IDs, 84 variants removed.\n",
      "Writing SAS/SAS_release9_remove_related_lrrk2_nodup.fam ... done.\n",
      "Writing SAS/SAS_release9_remove_related_lrrk2_nodup.bim ... done.\n",
      "Writing SAS/SAS_release9_remove_related_lrrk2_nodup.bed ... done.\n",
      "End time: Sun Apr  6 00:05:04 2025\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "# Remove duplicate\n",
    "ancestry_labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'FIN' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink2 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_lrrk2 \\\n",
    "    --rm-dup force-first \\\n",
    "    --make-bed \\\n",
    "    --out ${label}/${label}_release9_remove_related_lrrk2_nodup\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Haplotype block size comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AJ/AJ_release9_remove_related_lrrk2.blocks.log.\n",
      "Options in effect:\n",
      "  --bfile AJ/AJ_release9_remove_related_lrrk2_nodup\n",
      "  --blocks\n",
      "  --blocks-min-maf 0.0001\n",
      "  --out AJ/AJ_release9_remove_related_lrrk2.blocks\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "365 variants loaded from .bim file.\n",
      "3081 people (1934 males, 1147 females) loaded from .fam.\n",
      "2533 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3081 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997855.\n",
      "365 variants and 3081 people pass filters and QC.\n",
      "Among remaining phenotypes, 1709 are cases and 824 are controls.  (548\n",
      "phenotypes are missing.)\n",
      "--blocks: 13 haploblocks written to71818192020212122232324252526262728282930303131323333343535363637383839404041414243434445454646474848495050515152535354555556565758585960606161626363646565666667686869707071717273737475757676777878798080818182838384858586868788888990909191929393949595969697989899%\n",
      "AJ/AJ_release9_remove_related_lrrk2.blocks.blocks .\n",
      "Extra block details written to\n",
      "AJ/AJ_release9_remove_related_lrrk2.blocks.blocks.det .\n",
      "Longest span: 28.045kb.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_remove_related_lrrk2.blocks.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_lrrk2_nodup\n",
      "  --blocks\n",
      "  --blocks-min-maf 0.0001\n",
      "  --out CAH/CAH_release9_remove_related_lrrk2.blocks\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "364 variants loaded from .bim file.\n",
      "982 people (514 males, 468 females) loaded from .fam.\n",
      "954 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 982 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997863.\n",
      "364 variants and 982 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 310 are controls.  (28 phenotypes\n",
      "are missing.)\n",
      "--blocks: 20 haploblocks written to6171718181919202021212222232324252526262727282829293030313132323334343535363637373838393940404142424343444445454646474748484950505151525253535454555556565757585959606061616262636364646565666767686869697070717172727373747575767677777878797980808181828283848485858686878788888989909091929293939494959596969797989899%\n",
      "CAH/CAH_release9_remove_related_lrrk2.blocks.blocks .\n",
      "Extra block details written to\n",
      "CAH/CAH_release9_remove_related_lrrk2.blocks.blocks.det .\n",
      "Longest span: 24.35kb.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_release9_remove_related_lrrk2.blocks.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_lrrk2_nodup\n",
      "  --blocks\n",
      "  --blocks-min-maf 0.0001\n",
      "  --out CAS/CAS_release9_remove_related_lrrk2.blocks\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "362 variants loaded from .bim file.\n",
      "1006 people (461 males, 545 females) loaded from .fam.\n",
      "990 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1006 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997809.\n",
      "362 variants and 1006 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 329 are controls.  (16 phenotypes\n",
      "are missing.)\n",
      "--blocks: 13 haploblocks written to71818192020212122232324252526262728282929303131323233343435353637373839394040414242434344454546464748484950505151525353545455565657575859596060616262636464656566676768686970707171727373747575767677787879798081818282838484858586878788898990909192929393949595969697989899%\n",
      "CAS/CAS_release9_remove_related_lrrk2.blocks.blocks .\n",
      "Extra block details written to\n",
      "CAS/CAS_release9_remove_related_lrrk2.blocks.blocks.det .\n",
      "Longest span: 30.424kb.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_remove_related_lrrk2.blocks.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --blocks\n",
      "  --blocks-min-maf 0.0001\n",
      "  --out EAS/EAS_release9_remove_related_lrrk2.blocks\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "350 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998484.\n",
      "350 variants and 5646 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 2379 are controls.  (75\n",
      "phenotypes are missing.)\n",
      "--blocks: 15 haploblocks written to71819192021212222232424252626272828292930313132333334353536363738383940404142424343444545464747484949505051525253545455565657575859596061616263636464656666676868697070717172737374757576777778787980808182828384848585868787888989909191929293949495969697989899%\n",
      "EAS/EAS_release9_remove_related_lrrk2.blocks.blocks .\n",
      "Extra block details written to\n",
      "EAS/EAS_release9_remove_related_lrrk2.blocks.blocks.det .\n",
      "Longest span: 26.868kb.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_remove_related_lrrk2.blocks.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_lrrk2_nodup\n",
      "  --blocks\n",
      "  --blocks-min-maf 0.0001\n",
      "  --out EUR/EUR_release9_remove_related_lrrk2.blocks\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "332 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 34703 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998415.\n",
      "332 variants and 34703 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 5492 are controls.  (13879\n",
      "phenotypes are missing.)\n",
      "--blocks: 13 haploblocks written to71819192021212222232424252626272828292930313132333334353536363738383940404142424343444545464747484949505051525253545455565657575859596061616263636464656666676868697070717172737374757576777778787980808182828384848585868787888989909191929293949495969697989899%\n",
      "EUR/EUR_release9_remove_related_lrrk2.blocks.blocks .\n",
      "Extra block details written to\n",
      "EUR/EUR_release9_remove_related_lrrk2.blocks.blocks.det .\n",
      "Longest span: 22.805kb.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_lrrk2_nodup \\\n",
    "    --blocks \\\n",
    "    --blocks-min-maf 0.0001 \\\n",
    "    --out ${label}/${label}_release9_remove_related_lrrk2.blocks\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Create a table contain the location of A419V\n",
    "a419v_loci <- data.table(\n",
    "    SNP = \"A419V\" ,\n",
    "    CHR = 12 ,   \n",
    "    BP  = 40252984)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "   ancestry     KB NSNPS\n",
      "     <char>  <num> <int>\n",
      "1:     <NA>     NA    NA\n",
      "2:      EAS 14.193     9\n",
      "                                                                                                                                              SNPS\n",
      "                                                                                                                                            <char>\n",
      "1:                                                                                                                                            <NA>\n",
      "2: 12_40238792_G_A|12_40239262_G_A|12_40239512_A_G|12_40240543_G_A|12_40249666_G_A|12_40251369_G_A|12_40251495_G_A|12_40251828_G_A|12_40252984_G_A\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "# Initialize an empty table\n",
    "ALLhap <- data.table()\n",
    "\n",
    "ALLhap$ancestry <- NA\n",
    "ALLhap$KB       <- NA\n",
    "ALLhap$NSNPS    <- NA\n",
    "ALLhap$SNPS     <- NA\n",
    "\n",
    "# Extract info from the loci table into variables\n",
    "thisSnp <- a419v_loci$SNP[1]\n",
    "thisChr <- a419v_loci$CHR[1]\n",
    "thisBp  <- a419v_loci$BP[1]\n",
    "\n",
    "ancestry_labels <- c('AJ', 'CAH', 'CAS', 'EAS', 'EUR')\n",
    "\n",
    "for (anc in ancestry_labels){\n",
    "    \n",
    "    hap     <- fread(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_release9_remove_related_lrrk2.blocks.blocks.det\"), header =T)\n",
    "    hap_sub <- subset(hap, CHR == thisChr & BP1 <= thisBp & BP2 >= thisBp)\n",
    "    \n",
    "    if(length(hap_sub$KB) > 0) {\n",
    "    \n",
    "        tmp <- data.table(\n",
    "            \n",
    "            ancestry = anc,\n",
    "            KB       = hap_sub$KB,\n",
    "            NSNPS    = hap_sub$NSNPS,\n",
    "            SNPS     = hap_sub$SNPS\n",
    "        )\n",
    "        \n",
    "        ALLhap <- rbind(ALLhap, tmp)\n",
    "    \n",
    "    } \n",
    "}\n",
    "\n",
    "ALLhap"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Haplotype Association study"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Haplotype frequency comparison"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AJ/AJ_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile AJ/AJ_release9_remove_related_lrrk2_nodup\n",
      "  --out AJ/AJ_release9_remove_related_lrrk2_nodup\n",
      "  --recode\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "365 variants loaded from .bim file.\n",
      "3081 people (1934 males, 1147 females) loaded from .fam.\n",
      "2533 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3081 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997855.\n",
      "365 variants and 3081 people pass filters and QC.\n",
      "Among remaining phenotypes, 1709 are cases and 824 are controls.  (548\n",
      "phenotypes are missing.)\n",
      "--recode ped to AJ/AJ_release9_remove_related_lrrk2_nodup.ped +\n",
      "AJ/AJ_release9_remove_related_lrrk2_nodup.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_lrrk2_nodup\n",
      "  --out CAH/CAH_release9_remove_related_lrrk2_nodup\n",
      "  --recode\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "364 variants loaded from .bim file.\n",
      "982 people (514 males, 468 females) loaded from .fam.\n",
      "954 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 982 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997863.\n",
      "364 variants and 982 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 310 are controls.  (28 phenotypes\n",
      "are missing.)\n",
      "--recode ped to CAH/CAH_release9_remove_related_lrrk2_nodup.ped +\n",
      "CAH/CAH_release9_remove_related_lrrk2_nodup.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_lrrk2_nodup\n",
      "  --out CAS/CAS_release9_remove_related_lrrk2_nodup\n",
      "  --recode\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "362 variants loaded from .bim file.\n",
      "1006 people (461 males, 545 females) loaded from .fam.\n",
      "990 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 1006 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997809.\n",
      "362 variants and 1006 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 329 are controls.  (16 phenotypes\n",
      "are missing.)\n",
      "--recode ped to CAS/CAS_release9_remove_related_lrrk2_nodup.ped +\n",
      "CAS/CAS_release9_remove_related_lrrk2_nodup.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --out EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --recode\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "350 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998484.\n",
      "350 variants and 5646 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 2379 are controls.  (75\n",
      "phenotypes are missing.)\n",
      "--recode ped to EAS/EAS_release9_remove_related_lrrk2_nodup.ped +\n",
      "EAS/EAS_release9_remove_related_lrrk2_nodup.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_lrrk2_nodup\n",
      "  --out EUR/EUR_release9_remove_related_lrrk2_nodup\n",
      "  --recode\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "332 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 34703 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998415.\n",
      "332 variants and 34703 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 5492 are controls.  (13879\n",
      "phenotypes are missing.)\n",
      "--recode ped to EUR/EUR_release9_remove_related_lrrk2_nodup.ped +\n",
      "EUR/EUR_release9_remove_related_lrrk2_nodup.map ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_lrrk2_nodup \\\n",
    "    --recode \\\n",
    "    --out ${label}/${label}_release9_remove_related_lrrk2_nodup\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "ancestry_labels <- c('AJ', 'CAH', 'CAS', 'EAS', 'EUR')\n",
    "\n",
    "for (anc in ancestry_labels){\n",
    "    \n",
    "    # Read PED file\n",
    "    PED <- fread(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_release9_remove_related_lrrk2_nodup.ped\"))\n",
    "\n",
    "    # Get Allele name\n",
    "    FILTERED.LRRK2_1 <- read.table(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_release9_remove_related_lrrk2_nodup.map\"), header=FALSE)\n",
    "    FILTERED.LRRK2_2 <- read.table(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_release9_remove_related_lrrk2_nodup.map\"), header=FALSE)\n",
    "    FILTERED.LRRK2_1$V2 <- paste0(FILTERED.LRRK2_1$V2, sep=\"_\", \"1\")\n",
    "    FILTERED.LRRK2_2$V2 <- paste0(FILTERED.LRRK2_2$V2, sep=\"_\", \"2\")\n",
    "\n",
    "    # Set column names\n",
    "    colnames(FILTERED.LRRK2_1) <- c(\"CHR\", \"SNP\", \"CM\", \"POS\")\n",
    "    colnames(FILTERED.LRRK2_2) <- c(\"CHR\", \"SNP\", \"CM\", \"POS\")\n",
    "    FILTERED.LRRK2_2alleles    <- rbind(FILTERED.LRRK2_1, FILTERED.LRRK2_2)\n",
    "\n",
    "    # Sort file and create geno matrix:\n",
    "    # The variants should appear in order of genomic postition in the ped file, therefore, we sort the alleles based on position\n",
    "    LRRK2_pos     <- FILTERED.LRRK2_2alleles[order(FILTERED.LRRK2_2alleles$POS),]\n",
    "    LRRK2_alleles <- LRRK2_pos[,c(\"SNP\")]\n",
    "    write.table(LRRK2_alleles, file = paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_LRRK2_alleles.txt\"), quote = F, sep = \"\\t\", row.names = F, col.names = F)\n",
    "\n",
    "    # Add allele names to the PED file\n",
    "    colnames(PED) <- c(\"FID\", \"IID\", \"PAT\",\"MAT\", \"SEX\", \"PHENO\", LRRK2_alleles)\n",
    "    write.table(as.data.frame(PED), file = paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_geno_matrix_LRRK2.txt\"), quote = F, row.names = FALSE, sep = '\\t')\n",
    "\n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### EAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12_40238792_G_A|12_40239262_G_A|12_40239512_A_G|12_40240543_G_A|12_40249666_G_A|12_40251369_G_A|12_40251495_G_A|12_40251828_G_A|12_40252984_G_A\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"EAS\"\n",
    "\n",
    "grep 40252984 ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks | sed 's/ /|/g' | sed -r 's/^.{2}//'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "anc <- \"EAS\"\n",
    "\n",
    "Geno_matrix_LRRK2 <- read.delim(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_geno_matrix_LRRK2.txt\"), check.names = FALSE)\n",
    "\n",
    "#Colnames:\n",
    "LRRK2_alleles <- read.table(paste0(\"/home/jupyter/A419V_release9/\", anc, \"/\", anc, \"_LRRK2_alleles.txt\"), quote=\"\\\"\", comment.char=\"\")\n",
    "H1_a          <- LRRK2_alleles[grepl(\"12_40238792_G_A|12_40239262_G_A|12_40239512_A_G|12_40240543_G_A|12_40249666_G_A|12_40251369_G_A|12_40251495_G_A|12_40251828_G_A|12_40252984_G_A\", LRRK2_alleles$V1), ]\n",
    "H1_LRRK2      <- Geno_matrix_LRRK2[,c(\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENO\", H1_a)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] 24\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "ncol(H1_LRRK2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "ncol(H1_LRRK2)\n",
    "anc <- \"EAS\"\n",
    "\n",
    "#Extract only genotype data:\n",
    "geno          <- data.frame(H1_LRRK2[,c(7:24)], check.names = FALSE)\n",
    "\n",
    "#Set variables for running the association analyses in haplo.stats\n",
    "#Label the SNPs: \"12_40252984_G_A\"\n",
    "\n",
    "# Get the column name of the haplotype\n",
    "label <- colnames(H1_LRRK2)[7:24] %>% gsub(\"_.{1}$\", \"\", .) %>% unique()\n",
    "\n",
    "#Set binary pheno (0=control, 1=patient):\n",
    "H1_LRRK2$PHENO_01 <- H1_LRRK2$PHENO-1\n",
    "y.bin <- 1*(H1_LRRK2$PHENO_01==\"1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-------------------------------------------------------------------------------- \n",
      "                            Global Score Statistics                              \n",
      "-------------------------------------------------------------------------------- \n",
      "global-stat = 12, df = 4, p-val = 0.016\n",
      "\n",
      "-------------------------------------------------------------------------------- \n",
      "                         Counts for Cases and Controls                           \n",
      "-------------------------------------------------------------------------------- \n",
      "control    case \n",
      "   2454    3192 \n",
      "\n",
      "\n",
      "   12_40238792_G_A 12_40239262_G_A 12_40239512_A_G 12_40240543_G_A\n",
      "17               G               G               A               G\n",
      "21               G               G               G               G\n",
      "7                G               A               G               G\n",
      "4                A               G               A               G\n",
      "1                A               G               A               G\n",
      "2                A               G               A               G\n",
      "3                A               G               A               G\n",
      "5                A               G               G               G\n",
      "6                G               A               G               G\n",
      "8                G               G               A               G\n",
      "   12_40249666_G_A 12_40251369_G_A 12_40251495_G_A 12_40251828_G_A\n",
      "17               G               G               G               G\n",
      "21               A               G               G               A\n",
      "7                A               G               G               A\n",
      "4                G               G               G               G\n",
      "1                A               G               G               A\n",
      "2                G               G               G               A\n",
      "3                G               G               G               G\n",
      "5                A               G               G               A\n",
      "6                A               G               G               A\n",
      "8                A               G               G               A\n",
      "   12_40252984_G_A Hap-Score  p-val pool.hf control.hf case.hf glm.eff OR.lower\n",
      "17               G     -2.41 0.0161 0.31354    3.3e-01 0.30450     Eff     0.87\n",
      "21               G     -0.97 0.3315 0.35116    3.6e-01 0.34742    Base       NA\n",
      "7                G      0.98 0.3290 0.01258    1.1e-02 0.01348     Eff     0.86\n",
      "4                G      3.06 0.0022 0.29475    2.8e-01 0.30633     Eff     1.02\n",
      "1                G        NA     NA 0.00050    4.2e-04 0.00057    <NA>       NA\n",
      "2                G        NA     NA 0.00465    4.4e-03 0.00480       R     0.83\n",
      "3                A        NA     NA 0.00050         NA 0.00088    <NA>       NA\n",
      "5                G        NA     NA 0.00013    3.0e-04      NA    <NA>       NA\n",
      "6                A        NA     NA      NA    8.6e-05      NA    <NA>       NA\n",
      "8                G        NA     NA 0.00049    6.3e-04 0.00038    <NA>       NA\n",
      "     OR OR.upper\n",
      "17 0.96      1.1\n",
      "21 1.00       NA\n",
      "7  1.21      1.7\n",
      "4  1.12      1.2\n",
      "1    NA       NA\n",
      "2  1.07      1.4\n",
      "3    NA       NA\n",
      "5    NA       NA\n",
      "6    NA       NA\n",
      "8    NA       NA\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "#Non-adjusted:\n",
    "H1 <- haplo.cc(y=y.bin, geno=geno, locus.label= label, control = haplo.glm.control(haplo.freq.min = 0.01))\n",
    "print(H1, nlines=10, digits=2)\n",
    "\n",
    "#Sort the output on p-value:\n",
    "H1_cc.df <- H1$cc.df\n",
    "H1_cc.df_sort <- H1_cc.df[order(H1_cc.df$`p-val`),]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "|   |12_40238792_G_A |12_40239262_G_A |12_40239512_A_G |12_40240543_G_A |12_40249666_G_A |12_40251369_G_A |12_40251495_G_A |12_40251828_G_A |12_40252984_G_A |  Hap-Score|     p-val|   pool.hf| control.hf|   case.hf|glm.eff |  OR.lower|        OR| OR.upper|\n",
      "|:--|:---------------|:---------------|:---------------|:---------------|:---------------|:---------------|:---------------|:---------------|:---------------|----------:|---------:|---------:|----------:|---------:|:-------|---------:|---------:|--------:|\n",
      "|4  |A               |G               |A               |G               |G               |G               |G               |G               |G               |  3.0628115| 0.0021927| 0.2947496|  0.2797671| 0.3063289|Eff     | 1.0219151| 1.1215528| 1.230905|\n",
      "|17 |G               |G               |A               |G               |G               |G               |G               |G               |G               | -2.4072061| 0.0160751| 0.3135438|  0.3253539| 0.3045028|Eff     | 0.8731437| 0.9576553| 1.050347|\n",
      "|7  |G               |A               |G               |G               |A               |G               |G               |A               |G               |  0.9761736| 0.3289784| 0.0125783|  0.0113236| 0.0134770|Eff     | 0.8601473| 1.2117212| 1.706996|\n",
      "|21 |G               |G               |G               |G               |A               |G               |G               |A               |G               | -0.9710442| 0.3315263| 0.3511586|  0.3560257| 0.3474156|Base    |        NA| 1.0000000|       NA|\n",
      "|1  |A               |G               |A               |G               |A               |G               |G               |A               |G               |         NA|        NA| 0.0004956|  0.0004186| 0.0005659|NA      |        NA|        NA|       NA|\n",
      "|2  |A               |G               |A               |G               |G               |G               |G               |A               |G               |         NA|        NA| 0.0046469|  0.0043519| 0.0048033|R       | 0.8310355| 1.0680911| 1.372768|\n",
      "|3  |A               |G               |A               |G               |G               |G               |G               |G               |A               |         NA|        NA| 0.0004961|         NA| 0.0008805|NA      |        NA|        NA|       NA|\n",
      "|5  |A               |G               |G               |G               |A               |G               |G               |A               |G               |         NA|        NA| 0.0001329|  0.0003034|        NA|NA      |        NA|        NA|       NA|\n",
      "|6  |G               |A               |G               |G               |A               |G               |G               |A               |A               |         NA|        NA|        NA|  0.0000863|        NA|NA      |        NA|        NA|       NA|\n",
      "|8  |G               |G               |A               |G               |A               |G               |G               |A               |G               |         NA|        NA| 0.0004912|  0.0006265| 0.0003754|NA      |        NA|        NA|       NA|\n",
      "|9  |G               |G               |A               |G               |A               |G               |G               |G               |G               |         NA|        NA| 0.0000915|         NA| 0.0001616|NA      |        NA|        NA|       NA|\n",
      "|10 |G               |G               |A               |G               |G               |A               |A               |G               |A               |         NA|        NA|        NA|  0.0000000|        NA|NA      |        NA|        NA|       NA|\n",
      "|11 |G               |G               |A               |G               |G               |A               |A               |G               |G               |         NA|        NA|        NA|  0.0000000|        NA|NA      |        NA|        NA|       NA|\n",
      "|12 |G               |G               |A               |G               |G               |A               |G               |G               |A               |         NA|        NA|        NA|  0.0002152|        NA|NA      |        NA|        NA|       NA|\n",
      "|13 |G               |G               |A               |G               |G               |A               |G               |G               |G               |         NA|        NA| 0.0001779|  0.0000007| 0.0001568|NA      |        NA|        NA|       NA|\n",
      "|14 |G               |G               |A               |G               |G               |G               |A               |G               |G               |         NA|        NA| 0.0006203|  0.0004076| 0.0007840|NA      |        NA|        NA|       NA|\n",
      "|15 |G               |G               |A               |G               |G               |G               |G               |A               |G               |         NA|        NA| 0.0055149|  0.0071178| 0.0043280|R       | 0.8310355| 1.0680911| 1.372768|\n",
      "|16 |G               |G               |A               |G               |G               |G               |G               |G               |A               |         NA|        NA| 0.0082875|  0.0033735| 0.0119802|R       | 0.8310355| 1.0680911| 1.372768|\n",
      "|18 |G               |G               |G               |A               |A               |G               |G               |A               |G               |         NA|        NA| 0.0008865|  0.0010199| 0.0007839|NA      |        NA|        NA|       NA|\n",
      "|19 |G               |G               |G               |G               |A               |A               |G               |A               |G               |         NA|        NA| 0.0000000|  0.0000000|        NA|NA      |        NA|        NA|       NA|\n",
      "|20 |G               |G               |G               |G               |A               |G               |G               |A               |A               |         NA|        NA| 0.0000001|         NA| 0.0000017|NA      |        NA|        NA|       NA|\n",
      "|22 |G               |G               |G               |G               |A               |G               |G               |G               |G               |         NA|        NA| 0.0004475|  0.0008249| 0.0001570|NA      |        NA|        NA|       NA|\n",
      "|23 |G               |G               |G               |G               |G               |G               |G               |A               |G               |         NA|        NA| 0.0002953|  0.0003605| 0.0002726|NA      |        NA|        NA|       NA|\n",
      "|24 |G               |G               |G               |G               |G               |G               |G               |G               |G               |         NA|        NA| 0.0053855|  0.0084227| 0.0030249|R       | 0.8310355| 1.0680911| 1.372768|\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "library(knitr)\n",
    "\n",
    "H1_cc.df_sort %>% kable()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Visualise the block"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"/home/jupyter/A419V_release9/EAS/EAS_hap_extract.txt\", \"w\") as f:\n",
    "    f.write(\"12_40238792_G_A\\n12_40239262_G_A\\n12_40239512_A_G\\n12_40240543_G_A\\n12_40249666_G_A\\n12_40251369_G_A\\n12_40251495_G_A\\n12_40251828_G_A\\n12_40252984_G_A\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_remove_related_lrrk2_nodup.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --extract EAS/EAS_hap_extract.txt\n",
      "  --out EAS/EAS_release9_remove_related_lrrk2_nodup\n",
      "  --recode HV\n",
      "  --snps-only just-acgt\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "347 out of 350 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "--extract: 9 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 5646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.997816.\n",
      "9 variants and 5646 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 2379 are controls.  (75\n",
      "phenotypes are missing.)\n",
      "--recode HV to EAS/EAS_release9_remove_related_lrrk2_nodup.chr-*.ped + .info...\n",
      "[chromosome 12              done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"EAS\"\n",
    "\n",
    "/home/jupyter/plink1.9 \\\n",
    "--bfile ${label}/${label}_release9_remove_related_lrrk2_nodup \\\n",
    "--extract ${label}/${label}_hap_extract.txt \\\n",
    "--snps-only just-acgt \\\n",
    "--recode HV \\\n",
    "--out ${label}/${label}_release9_remove_related_lrrk2_nodup"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.info\n",
      "EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.ped\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "\n",
    "ls EAS/EAS_release9_remove_related_lrrk2_nodup.chr*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing: gsutil -u terra-8cb3be5c -m cp /home/jupyter/A419V_release9/EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.* gs://fc-e8a73e41-545c-42b1-8720-970cf953ba35/A419V_analysis/release9/haplotype/EAS/\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Copying file:///home/jupyter/A419V_release9/EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.ped [Content-Type=application/octet-stream]...\n",
      "Copying file:///home/jupyter/A419V_release9/EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.info [Content-Type=application/x-info]...\n",
      "/ [2/2 files][336.2 KiB/336.2 KiB] 100% Done                                    \n",
      "Operation completed over 2 objects/336.2 KiB.                                    \n"
     ]
    }
   ],
   "source": [
    "WORK_DIR=\"/home/jupyter/A419V_release9\"\n",
    "shell_do(f'gsutil -u {BILLING_PROJECT_ID} -m cp {WORK_DIR}/EAS/EAS_release9_remove_related_lrrk2_nodup.chr-12.* {WORKSPACE_BUCKET}/A419V_analysis/release9/haplotype/EAS/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### EUR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "* 12_40198826_A_G 12_40199601_A_G 12_40200768_A_G 12_40204350_A_C 12_40207140_A_G 12_40208138_G_A 12_40208422_C_A 12_40209874_C_A 12_40213423_A_T 12_40215078_G_A 12_40215707_A_G 12_40216942_G_A 12_40220260_A_G 12_40221630_A_G\n",
      "* 12_40225280_A_G 12_40225499_A_G\n",
      "* 12_40237989_G_A 12_40238792_G_A\n",
      "* 12_40256700_G_A 12_40257073_T_A 12_40258718_G_A\n",
      "* 12_40262377_A_G 12_40262593_G_A 12_40263898_C_G 12_40264353_G_A\n",
      "* 12_40267251_T_A 12_40272118_G_A 12_40273073_G_C 12_40273861_G_A 12_40276652_A_G 12_40277068_A_G 12_40277196_C_A 12_40278187_A_G 12_40279566_A_G 12_40282999_C_A 12_40283045_A_T 12_40283227_C_A 12_40284011_C_A 12_40285467_T_A 12_40285609_G_A 12_40287115_A_G\n",
      "* 12_40293043_A_G 12_40293294_G_A 12_40293624_G_C 12_40293896_A_G 12_40294893_A_G 12_40295786_G_A\n",
      "* 12_40309109_G_A 12_40309185_G_A\n",
      "* 12_40311356_A_T 12_40312802_G_A 12_40313594_C_A 12_40313976_G_A 12_40314059_G_A 12_40317244_A_G 12_40317645_C_A 12_40317832_G_A 12_40320043_C_G 12_40320097_A_G 12_40320099_T_A 12_40320699_C_A 12_40322047_C_A 12_40322386_C_A 12_40322458_G_A 12_40322632_A_G 12_40322672_A_G\n",
      "* 12_40328650_A_G 12_40329673_T_A 12_40330228_G_A\n",
      "* 12_40331158_C_A 12_40331976_A_G\n",
      "* 12_40338321_A_G 12_40338928_G_C 12_40339034_A_G 12_40339098_A_C 12_40340400_G_A 12_40341862_A_G 12_40342603_A_G 12_40343200_A_G 12_40343381_A_G 12_40343438_A_G 12_40344001_G_A 12_40346563_C_A 12_40346830_I_D 12_40346884_A_G 12_40347071_A_G 12_40348019_G_A 12_40348204_C_A 12_40348452_G_A 12_40348859_G_A 12_40348899_G_A 12_40350343_A_C 12_40350973_A_C 12_40351006_G_A 12_40351379_G_A\n",
      "* 12_40352036_A_G 12_40352153_A_G 12_40357725_G_A 12_40359994_A_G 12_40361487_A_G 12_40362670_A_C 12_40363440_G_A 12_40363528_A_G 12_40364489_G_A 12_40364850_G_A 12_40365793_G_A 12_40366046_T_A 12_40366829_A_G 12_40367205_G_A\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"EUR\"\n",
    "\n",
    "cat ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"EUR\"\n",
    "\n",
    "grep 40252984 ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks | sed 's/ /|/g' | sed -r 's/^.{2}//'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "No block contain p.A419V, checking for missingness"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_remove_related_lrrk2.blocks_test.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_lrrk2_nodup\n",
      "  --missing\n",
      "  --out EUR/EUR_release9_remove_related_lrrk2.blocks_test\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "332 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 34703 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.998415.\n",
      "--missing: Sample missing data report written to\n",
      "EUR/EUR_release9_remove_related_lrrk2.blocks_test.imiss, and variant-based\n",
      "missing data report written to\n",
      "EUR/EUR_release9_remove_related_lrrk2.blocks_test.lmiss.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_lrrk2_nodup \\\n",
    "    --missing \\\n",
    "    --out ${label}/${label}_release9_remove_related_lrrk2.blocks_test\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR               SNP   N_MISS   N_GENO   F_MISS\n",
      "  12   12_40252984_G_A       35    34703 0.001009\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "head -1 EUR/EUR_release9_remove_related_lrrk2.blocks_test.lmiss\n",
    "grep 40252984 EUR/EUR_release9_remove_related_lrrk2.blocks_test.lmiss"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AJ"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"AJ\"\n",
    "\n",
    "grep 40252984 ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks | sed 's/ /|/g' | sed -r 's/^.{2}//'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "* 12_40197810_C_A 12_40198586_A_G 12_40198826_A_G 12_40199601_A_G 12_40200768_A_G 12_40204350_A_C 12_40207140_A_G 12_40208138_G_A\n",
      "* 12_40208422_C_A 12_40209874_C_A 12_40213423_A_T 12_40213907_A_G 12_40215078_A_G 12_40215707_A_G 12_40216942_G_A 12_40217062_C_G 12_40220260_A_G 12_40220632_G_A 12_40220854_G_A 12_40221630_A_G 12_40222612_A_G 12_40223957_A_G 12_40226614_T_A 12_40227006_G_C 12_40228666_C_A 12_40230508_A_G\n",
      "* 12_40237989_G_A 12_40238792_G_A 12_40239512_A_G 12_40249666_G_A\n",
      "* 12_40251828_A_G 12_40252732_G_A\n",
      "* 12_40257073_T_A 12_40258718_G_A 12_40261071_A_G 12_40262377_A_G 12_40262593_G_A 12_40263898_C_G 12_40264353_G_A\n",
      "* 12_40267251_T_A 12_40272118_G_A 12_40273073_G_C 12_40273861_G_A 12_40276592_G_A 12_40277196_C_A 12_40278187_A_G 12_40279566_A_G 12_40282999_C_A 12_40283045_A_T 12_40283227_C_A 12_40285467_T_A 12_40285609_G_A 12_40287115_A_G 12_40288195_A_G 12_40288634_G_A 12_40293043_A_G\n",
      "* 12_40293896_A_G 12_40294893_A_G 12_40295786_G_A 12_40296180_A_G 12_40298346_G_A 12_40299255_A_G 12_40300597_G_A\n",
      "* 12_40303632_G_A 12_40303675_G_A\n",
      "* 12_40307469_C_A 12_40309044_T_A 12_40309109_G_A 12_40309185_G_A 12_40310434_G_A 12_40310652_A_G 12_40311356_A_T 12_40312802_G_A 12_40313594_C_A 12_40313976_G_A 12_40314059_G_A 12_40315611_A_T 12_40317244_A_G 12_40317645_C_A 12_40317832_G_A 12_40318391_A_G 12_40319758_C_A 12_40320032_A_C 12_40320043_G_A 12_40320071_G_A 12_40320097_A_G 12_40320099_T_A 12_40320386_A_G 12_40320699_C_A 12_40322386_G_A 12_40322458_G_A\n",
      "* 12_40322632_A_G 12_40322672_A_G 12_40328650_A_G 12_40329673_T_A 12_40330228_G_A\n",
      "* 12_40331158_C_A 12_40331976_A_G\n",
      "* 12_40334626_C_G 12_40336661_G_A 12_40338321_A_G 12_40338928_G_C 12_40339034_A_G 12_40339098_A_C 12_40340400_G_A 12_40341862_A_G 12_40342447_G_A 12_40342603_A_G 12_40342980_A_G 12_40343200_A_G 12_40343381_A_G 12_40343438_A_G 12_40344001_G_A 12_40346563_C_A 12_40346884_A_G 12_40347071_A_G 12_40348019_G_A 12_40348203_G_A 12_40348204_C_A 12_40348271_A_G 12_40348452_G_A 12_40348899_G_A 12_40350284_T_A 12_40350343_A_C 12_40350381_A_G 12_40350973_A_C 12_40351006_G_A 12_40351379_G_A 12_40352036_A_G 12_40352153_A_G 12_40357725_G_A 12_40359994_A_G 12_40361487_A_G 12_40362670_A_C\n",
      "* 12_40363528_A_G 12_40364489_G_A 12_40364850_G_A 12_40364960_A_G 12_40365793_G_A 12_40366046_T_A 12_40366829_A_G 12_40367044_G_A 12_40367205_G_A 12_40368744_G_A\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"AJ\"\n",
    "\n",
    "cat ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "No block contain p.A419V"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CAH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"CAH\"\n",
    "\n",
    "grep 40252984 ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks | sed 's/ /|/g' | sed -r 's/^.{2}//'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "* 12_40199601_A_G 12_40200768_A_G 12_40204350_A_C 12_40207140_A_G 12_40208138_G_A 12_40208422_C_A 12_40209874_C_A 12_40213423_A_T 12_40213907_A_G 12_40215078_G_A 12_40215707_A_G\n",
      "* 12_40216942_G_A 12_40217062_C_G\n",
      "* 12_40220854_G_A 12_40221630_A_G 12_40222612_A_G\n",
      "* 12_40225280_A_G 12_40225499_A_G\n",
      "* 12_40227006_G_C 12_40228666_C_A\n",
      "* 12_40237989_A_G 12_40238792_G_A 12_40239512_A_G 12_40240543_G_A 12_40243579_A_C 12_40247932_A_T 12_40249666_G_A\n",
      "* 12_40251828_G_A 12_40252732_G_A\n",
      "* 12_40257073_T_A 12_40258718_G_A 12_40261071_A_G 12_40261443_G_A 12_40262377_A_G 12_40262593_G_A 12_40263898_C_G 12_40264353_G_A\n",
      "* 12_40272118_G_A 12_40273073_G_C\n",
      "* 12_40277196_C_A 12_40278187_A_G 12_40279566_A_G 12_40282999_C_A 12_40283045_A_T 12_40283227_C_A 12_40283897_G_A 12_40283933_G_A 12_40284011_C_A 12_40285467_T_A\n",
      "* 12_40287115_A_G 12_40288195_A_G 12_40288634_G_A 12_40293043_A_G 12_40293294_G_A 12_40293552_A_C 12_40293624_G_C 12_40293896_A_G 12_40294893_A_G 12_40295280_G_A 12_40295786_G_A\n",
      "* 12_40303632_G_A 12_40303675_G_A\n",
      "* 12_40309044_T_A 12_40309109_G_A\n",
      "* 12_40309185_G_A 12_40310434_G_A 12_40310504_C_G 12_40310652_A_G 12_40311356_A_T 12_40312802_G_A 12_40313594_C_A\n",
      "* 12_40315611_A_T 12_40317244_A_G 12_40317645_C_A 12_40317832_G_A 12_40318391_A_G 12_40319758_C_A 12_40320032_A_C 12_40320043_C_G 12_40320071_G_A 12_40320097_A_G 12_40320099_T_A 12_40320386_A_G 12_40320699_C_A 12_40321181_A_G 12_40322458_G_A 12_40322632_A_G 12_40322672_A_G\n",
      "* 12_40328650_A_G 12_40329673_T_A 12_40330228_G_A\n",
      "* 12_40331158_C_A 12_40331976_A_G\n",
      "* 12_40334626_C_G 12_40336661_G_A\n",
      "* 12_40338321_A_G 12_40338928_G_C 12_40339034_A_G 12_40339098_C_A 12_40340400_G_A 12_40341862_A_G 12_40342447_G_A 12_40342603_A_G 12_40342980_A_G 12_40343200_A_G 12_40343381_A_G 12_40343438_A_G 12_40344001_G_A 12_40346563_C_A 12_40346884_A_G 12_40347071_A_G 12_40348019_G_A 12_40348203_G_A 12_40348204_A_C 12_40348271_A_G 12_40348452_G_A 12_40348859_G_A 12_40348899_G_A 12_40350284_T_A 12_40350343_A_C 12_40350381_A_G 12_40350973_A_C 12_40351006_G_A 12_40351379_G_A 12_40351579_G_A 12_40351723_A_G 12_40352036_A_G 12_40352153_A_G 12_40357725_G_A 12_40359994_A_G 12_40361487_A_G 12_40362670_A_C\n",
      "* 12_40363528_A_G 12_40364489_G_A 12_40364850_G_A 12_40365793_G_A 12_40366046_T_A 12_40366829_A_G 12_40367205_G_A 12_40367664_G_A 12_40368744_G_A\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"CAH\"\n",
    "\n",
    "cat ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### CAS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=\"/home/jupyter/A419V_release9/\"\n",
    "cd $WORK_DIR\n",
    "\n",
    "label=\"CAS\"\n",
    "\n",
    "grep 40252984 ${label}/${label}_release9_remove_related_lrrk2.blocks.blocks | sed 's/ /|/g' | sed -r 's/^.{2}//'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate R2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "162 AJ/AJ_release9_remove_related_lrrk2_nodup_exon.bim\n",
      "162 CAH/CAH_release9_remove_related_lrrk2_nodup_exon.bim\n",
      "162 CAS/CAS_release9_remove_related_lrrk2_nodup_exon.bim\n",
      "161 EAS/EAS_release9_remove_related_lrrk2_nodup_exon.bim\n",
      "160 EUR/EUR_release9_remove_related_lrrk2_nodup_exon.bim\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    wc -l ${label}/${label}_release9_remove_related_lrrk2_nodup_exon.bim\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to AJ/AJ_a419v_r2.log.\n",
      "Options in effect:\n",
      "  --bfile AJ/AJ_release9_remove_related_lrrk2_nodup_exon\n",
      "  --ld-snp 12_40252984_G_A\n",
      "  --ld-window 99999\n",
      "  --ld-window-kb 10000000\n",
      "  --ld-window-r2 0\n",
      "  --out AJ/AJ_a419v_r2\n",
      "  --r2\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "162 variants loaded from .bim file.\n",
      "3081 people (1934 males, 1147 females) loaded from .fam.\n",
      "2533 phenotype values loaded from .fam.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 3081 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999088.\n",
      "162 variants and 3081 people pass filters and QC.\n",
      "Among remaining phenotypes, 1709 are cases and 824 are controls.  (548\n",
      "phenotypes are missing.)\n",
      "--r2 to AJ/AJ_a419v_r2.ld ... 0% [processingwriting]          done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_a419v_r2.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_lrrk2_nodup_exon\n",
      "  --ld-snp 12_40252984_G_A\n",
      "  --ld-window 99999\n",
      "  --ld-window-kb 10000000\n",
      "  --ld-window-r2 0\n",
      "  --out CAH/CAH_a419v_r2\n",
      "  --r2\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "162 variants loaded from .bim file.\n",
      "982 people (514 males, 468 females) loaded from .fam.\n",
      "954 phenotype values loaded from .fam.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 982 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999126.\n",
      "162 variants and 982 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 310 are controls.  (28 phenotypes\n",
      "are missing.)\n",
      "--r2 to CAH/CAH_a419v_r2.ld ... 0% [processingwriting]          done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_a419v_r2.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_lrrk2_nodup_exon\n",
      "  --ld-snp 12_40252984_G_A\n",
      "  --ld-window 99999\n",
      "  --ld-window-kb 10000000\n",
      "  --ld-window-r2 0\n",
      "  --out CAS/CAS_a419v_r2\n",
      "  --r2\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "162 variants loaded from .bim file.\n",
      "1006 people (461 males, 545 females) loaded from .fam.\n",
      "990 phenotype values loaded from .fam.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 1006 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999202.\n",
      "162 variants and 1006 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 329 are controls.  (16 phenotypes\n",
      "are missing.)\n",
      "--r2 to CAS/CAS_a419v_r2.ld ... 0% [processingwriting]          done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_a419v_r2.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_lrrk2_nodup_exon\n",
      "  --ld-snp 12_40252984_G_A\n",
      "  --ld-window 99999\n",
      "  --ld-window-kb 10000000\n",
      "  --ld-window-r2 0\n",
      "  --out EAS/EAS_a419v_r2\n",
      "  --r2\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "161 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 5646 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999303.\n",
      "161 variants and 5646 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 2379 are controls.  (75\n",
      "phenotypes are missing.)\n",
      "--r2 to EAS/EAS_a419v_r2.ld ... 0% [processingwriting]          done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_a419v_r2.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_lrrk2_nodup_exon\n",
      "  --ld-snp 12_40252984_G_A\n",
      "  --ld-window 99999\n",
      "  --ld-window-kb 10000000\n",
      "  --ld-window-r2 0\n",
      "  --out EUR/EUR_a419v_r2\n",
      "  --r2\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "160 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "Using up to 8 threads (change this with --threads).\n",
      "Before main variant filters, 34703 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999231.\n",
      "160 variants and 34703 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 5492 are controls.  (13879\n",
      "phenotypes are missing.)\n",
      "--r2 to EUR/EUR_a419v_r2.ld ... 0% [processingwriting]          done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_lrrk2_nodup_exon \\\n",
    "    --ld-snp \"12_40252984_G_A\" \\\n",
    "    --r2 \\\n",
    "    --ld-window 99999 \\\n",
    "    --ld-window-kb 10000000 \\\n",
    "    --ld-window-r2 0 \\\n",
    "    --out ${label}/${label}_a419v_r2\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "23 AJ/AJ_a419v_r2.ld\n",
      "37 CAH/CAH_a419v_r2.ld\n",
      "28 CAS/CAS_a419v_r2.ld\n",
      "63 EAS/EAS_a419v_r2.ld\n",
      "111 EUR/EUR_a419v_r2.ld\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\"\n",
    "do\n",
    "\n",
    "    wc -l ${label}/${label}_a419v_r2.ld\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "pd.set_option(\"Display.max_rows\", None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_85/1821464278.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n",
      "/tmp/ipykernel_85/1821464278.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n",
      "/tmp/ipykernel_85/1821464278.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n",
      "/tmp/ipykernel_85/1821464278.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n",
      "/tmp/ipykernel_85/1821464278.py:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP_B</th>\n",
       "      <th>R2_AJ</th>\n",
       "      <th>R2_CAH</th>\n",
       "      <th>R2_CAS</th>\n",
       "      <th>R2_EAS</th>\n",
       "      <th>R2_EUR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>12_40232380_A_C</td>\n",
       "      <td>2.118590e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.431000e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029200e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>12_40237989_G_A</td>\n",
       "      <td>1.765370e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.127060e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>12_40251273_G_A</td>\n",
       "      <td>3.172710e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.699050e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>12_40251369_G_A</td>\n",
       "      <td>1.059640e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.214590e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>12_40252984_G_A</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>12_40263898_C_G</td>\n",
       "      <td>6.132380e-05</td>\n",
       "      <td>0.002827</td>\n",
       "      <td>5.127230e-07</td>\n",
       "      <td>0.000534</td>\n",
       "      <td>1.018600e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>12_40278187_A_G</td>\n",
       "      <td>9.408870e-05</td>\n",
       "      <td>0.001466</td>\n",
       "      <td>2.057570e-03</td>\n",
       "      <td>0.001594</td>\n",
       "      <td>6.576440e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>12_40293624_G_C</td>\n",
       "      <td>2.122730e-07</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.339440e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>12_40294893_A_G</td>\n",
       "      <td>1.132000e-03</td>\n",
       "      <td>0.047171</td>\n",
       "      <td>6.784000e-02</td>\n",
       "      <td>0.043004</td>\n",
       "      <td>3.787680e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>12_40298346_G_A</td>\n",
       "      <td>1.056880e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12_40299255_A_G</td>\n",
       "      <td>1.059640e-07</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>1.720700e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.151250e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12_40309109_G_A</td>\n",
       "      <td>5.873980e-05</td>\n",
       "      <td>0.002641</td>\n",
       "      <td>9.049600e-06</td>\n",
       "      <td>0.000479</td>\n",
       "      <td>1.045930e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>12_40310434_G_A</td>\n",
       "      <td>1.058260e-07</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>8.491320e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>12_40313976_G_A</td>\n",
       "      <td>1.702630e-06</td>\n",
       "      <td>0.000069</td>\n",
       "      <td>5.151640e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.825750e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>12_40314059_G_A</td>\n",
       "      <td>1.293960e-05</td>\n",
       "      <td>0.000366</td>\n",
       "      <td>5.670420e-04</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>4.627960e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>12_40320043_G_A</td>\n",
       "      <td>1.058260e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>1.819250e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>12_40320097_A_G</td>\n",
       "      <td>1.642700e-05</td>\n",
       "      <td>0.000333</td>\n",
       "      <td>1.987480e-04</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.945370e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>12_40320099_T_A</td>\n",
       "      <td>7.229820e-05</td>\n",
       "      <td>0.000042</td>\n",
       "      <td>6.789790e-03</td>\n",
       "      <td>0.005020</td>\n",
       "      <td>3.587880e-04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>12_40340400_G_A</td>\n",
       "      <td>1.233360e-03</td>\n",
       "      <td>0.000209</td>\n",
       "      <td>3.431000e-05</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.677650e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>12_40346884_A_G</td>\n",
       "      <td>3.395590e-05</td>\n",
       "      <td>0.000482</td>\n",
       "      <td>9.249970e-04</td>\n",
       "      <td>0.000043</td>\n",
       "      <td>2.247010e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>12_40364850_G_A</td>\n",
       "      <td>5.602140e-05</td>\n",
       "      <td>0.003045</td>\n",
       "      <td>1.041340e-02</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.274230e-03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>12_40364960_A_G</td>\n",
       "      <td>2.114460e-07</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.031130e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>12_40232383_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000110</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.073290e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>12_40235634_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5.146180e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>12_40237989_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.001524</td>\n",
       "      <td>1.694470e-02</td>\n",
       "      <td>0.004534</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>12_40240543_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>2.427070e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>12_40243579_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030780e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>12_40274666_A_T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000055</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.067330e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>12_40283897_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>5.151640e-05</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>3.030600e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>12_40283933_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.075979</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030780e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>12_40284011_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>1.717240e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.639500e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>12_40293552_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000028</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.064530e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>12_40302866_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.949580e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>12_40305791_C_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000111</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.126820e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>12_40308618_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>1.037600e-04</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.097200e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>12_40308672_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>2.121050e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>12_40309145_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000083</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.821460e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>12_40310504_C_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>12_40351579_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000125</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.069080e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>12_40351723_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.550370e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>12_40363440_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.678690e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>12_40363526_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>1.911850e-04</td>\n",
       "      <td>0.000024</td>\n",
       "      <td>9.091280e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>12_40367664_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>3.431000e-05</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>2.726910e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>12_40293616_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029900e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>12_40293629_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.206870e-04</td>\n",
       "      <td>0.000052</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>12_40310450_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>0.000067</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>12_40323256_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7.279530e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>12_40359345_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.713790e-05</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.515130e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>12_40237990_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>12_40249859_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>12_40251495_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000021</td>\n",
       "      <td>3.030950e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>12_40259538_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.031390e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>12_40274899_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.031480e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>12_40278002_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>12_40283985_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>12_40284008_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>12_40287381_T_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.032880e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>12_40295568_G_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.031480e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>12_40299173_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.030950e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>12_40299212_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.818830e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>12_40304000_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>2.121790e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>12_40308624_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.065930e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>12_40308669_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000040</td>\n",
       "      <td>3.032700e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>12_40309153_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>12_40310434_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>12_40310435_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>9.091280e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>12_40310452_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>12_40310461_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>12_40310515_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.070840e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>12_40320036_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.063830e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>12_40322038_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>2.947880e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>12_40322047_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>2.425810e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>12_40322130_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>12_40322145_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.030250e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>12_40322322_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>12_40322468_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>6.073640e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>12_40323255_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.036560e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>12_40340380_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>12_40351572_T_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>12_40354474_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>12_40359344_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.002956</td>\n",
       "      <td>1.515040e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>12_40359354_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>9.094950e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>12_40363409_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>12_40363541_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>9.101780e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>12_40364850_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.003586</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>12_40367045_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>3.032880e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>12_40225159_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.033050e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>12_40235700_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.034100e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>12_40240485_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.031650e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>12_40251361_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.031130e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>12_40251471_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029200e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>12_40252990_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.077160e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>12_40257334_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030780e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>12_40263875_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029900e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>12_40274661_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.031130e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>12_40278154_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.059630e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>12_40284011_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.033050e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>12_40284061_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.061380e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>12_40284113_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.061030e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>12_40284132_I_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.821780e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>12_40287438_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.077510e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>12_40294866_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030600e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>103</th>\n",
       "      <td>12_40294879_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030950e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>12_40295466_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030600e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>105</th>\n",
       "      <td>12_40295567_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.058930e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>12_40299216_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030080e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>107</th>\n",
       "      <td>12_40304040_G_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.123400e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>12_40308481_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.190180e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>12_40308525_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.032350e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>12_40309225_A_C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.121670e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>12_40309226_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.076100e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>12_40310434_C_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.794320e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>12_40310561_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.093380e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>12_40310635_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030430e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>115</th>\n",
       "      <td>12_40314047_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.064880e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>116</th>\n",
       "      <td>12_40320013_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.032880e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>117</th>\n",
       "      <td>12_40320040_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.033580e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>12_40321114_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.217330e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>119</th>\n",
       "      <td>12_40322047_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.960340e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>12_40322131_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030080e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>12_40323221_I_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029730e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>12_40323270_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.073290e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>12_40328362_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.030430e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>12_40335031_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.797880e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>12_40335091_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.107570e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>12_40340361_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.029900e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>127</th>\n",
       "      <td>12_40340404_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.820940e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>128</th>\n",
       "      <td>12_40346830_I_D</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.193500e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>12_40351585_G_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.950720e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>12_40359338_A_G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.060680e-08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>131</th>\n",
       "      <td>12_40363540_C_A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.095480e-08</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               SNP_B         R2_AJ    R2_CAH        R2_CAS    R2_EAS  \\\n",
       "0                              NaN       NaN           NaN       NaN   \n",
       "1    12_40232380_A_C  2.118590e-07       NaN  3.431000e-05       NaN   \n",
       "2    12_40237989_G_A  1.765370e-05       NaN           NaN       NaN   \n",
       "3    12_40251273_G_A  3.172710e-07       NaN           NaN       NaN   \n",
       "4    12_40251369_G_A  1.059640e-07       NaN           NaN  0.000003   \n",
       "5    12_40252984_G_A  1.000000e+00  1.000000  1.000000e+00  1.000000   \n",
       "6    12_40263898_C_G  6.132380e-05  0.002827  5.127230e-07  0.000534   \n",
       "7    12_40278187_A_G  9.408870e-05  0.001466  2.057570e-03  0.001594   \n",
       "8    12_40293624_G_C  2.122730e-07  0.000014           NaN       NaN   \n",
       "9    12_40294893_A_G  1.132000e-03  0.047171  6.784000e-02  0.043004   \n",
       "10   12_40298346_G_A  1.056880e-07       NaN           NaN  0.000012   \n",
       "11   12_40299255_A_G  1.059640e-07  0.000041  1.720700e-05       NaN   \n",
       "12   12_40309109_G_A  5.873980e-05  0.002641  9.049600e-06  0.000479   \n",
       "13   12_40310434_G_A  1.058260e-07  0.000014  1.713790e-05  0.000012   \n",
       "14   12_40313976_G_A  1.702630e-06  0.000069  5.151640e-05       NaN   \n",
       "15   12_40314059_G_A  1.293960e-05  0.000366  5.670420e-04  0.000012   \n",
       "16   12_40320043_G_A  1.058260e-07       NaN           NaN  0.000006   \n",
       "17   12_40320097_A_G  1.642700e-05  0.000333  1.987480e-04       NaN   \n",
       "18   12_40320099_T_A  7.229820e-05  0.000042  6.789790e-03  0.005020   \n",
       "19   12_40340400_G_A  1.233360e-03  0.000209  3.431000e-05  0.000003   \n",
       "20   12_40346884_A_G  3.395590e-05  0.000482  9.249970e-04  0.000043   \n",
       "21   12_40364850_G_A  5.602140e-05  0.003045  1.041340e-02       NaN   \n",
       "22   12_40364960_A_G  2.114460e-07       NaN           NaN       NaN   \n",
       "23   12_40232383_A_G           NaN  0.000110           NaN  0.000003   \n",
       "24   12_40235634_A_G           NaN  0.000027           NaN       NaN   \n",
       "25   12_40237989_A_G           NaN  0.001524  1.694470e-02  0.004534   \n",
       "26   12_40240543_G_A           NaN  0.000028           NaN  0.000031   \n",
       "27   12_40243579_A_C           NaN  0.000041           NaN       NaN   \n",
       "28   12_40274666_A_T           NaN  0.000055           NaN       NaN   \n",
       "29   12_40283897_G_A           NaN  0.000027  5.151640e-05  0.000002   \n",
       "30   12_40283933_G_A           NaN  0.075979           NaN       NaN   \n",
       "31   12_40284011_C_A           NaN  0.000014  1.717240e-05       NaN   \n",
       "32   12_40293552_A_C           NaN  0.000028           NaN       NaN   \n",
       "33   12_40302866_A_G           NaN  0.000014           NaN       NaN   \n",
       "34   12_40305791_C_G           NaN  0.000111           NaN       NaN   \n",
       "35   12_40308618_A_G           NaN  0.000014  1.037600e-04  0.000003   \n",
       "36   12_40308672_G_A           NaN  0.000014           NaN  0.000003   \n",
       "37   12_40309145_G_A           NaN  0.000083  1.713790e-05       NaN   \n",
       "38   12_40310504_C_G           NaN  0.000013           NaN       NaN   \n",
       "39   12_40351579_G_A           NaN  0.000125           NaN       NaN   \n",
       "40   12_40351723_A_G           NaN  0.000014           NaN       NaN   \n",
       "41   12_40363440_G_A           NaN  0.000014           NaN       NaN   \n",
       "42   12_40363526_G_A           NaN  0.000041  1.911850e-04  0.000024   \n",
       "43   12_40367664_G_A           NaN  0.000027  3.431000e-05  0.000009   \n",
       "44   12_40293616_G_A           NaN       NaN  1.713790e-05       NaN   \n",
       "45   12_40293629_G_A           NaN       NaN  1.206870e-04  0.000052   \n",
       "46   12_40310450_G_A           NaN       NaN  1.713790e-05  0.000067   \n",
       "47   12_40323256_A_G           NaN       NaN  1.713790e-05       NaN   \n",
       "48   12_40359345_G_A           NaN       NaN  1.713790e-05  0.000003   \n",
       "49   12_40237990_A_C           NaN       NaN           NaN  0.000006   \n",
       "50   12_40249859_A_G           NaN       NaN           NaN  0.000003   \n",
       "51   12_40251495_G_A           NaN       NaN           NaN  0.000021   \n",
       "52   12_40259538_G_A           NaN       NaN           NaN  0.000003   \n",
       "53   12_40274899_A_G           NaN       NaN           NaN  0.000003   \n",
       "54   12_40278002_G_A           NaN       NaN           NaN  0.000003   \n",
       "55   12_40283985_C_A           NaN       NaN           NaN  0.000003   \n",
       "56   12_40284008_G_A           NaN       NaN           NaN  0.000003   \n",
       "57   12_40287381_T_A           NaN       NaN           NaN  0.000003   \n",
       "58   12_40295568_G_C           NaN       NaN           NaN  0.000003   \n",
       "59   12_40299173_A_G           NaN       NaN           NaN  0.000003   \n",
       "60   12_40299212_G_A           NaN       NaN           NaN  0.000003   \n",
       "61   12_40304000_G_A           NaN       NaN           NaN  0.000003   \n",
       "62   12_40308624_G_A           NaN       NaN           NaN  0.000003   \n",
       "63   12_40308669_C_A           NaN       NaN           NaN  0.000040   \n",
       "64   12_40309153_G_A           NaN       NaN           NaN  0.000037   \n",
       "65   12_40310434_C_A           NaN       NaN           NaN  0.000003   \n",
       "66   12_40310435_G_A           NaN       NaN           NaN  0.000006   \n",
       "67   12_40310452_G_A           NaN       NaN           NaN  0.000003   \n",
       "68   12_40310461_G_A           NaN       NaN           NaN  0.000006   \n",
       "69   12_40310515_A_G           NaN       NaN           NaN  0.000003   \n",
       "70   12_40320036_A_G           NaN       NaN           NaN  0.000003   \n",
       "71   12_40322038_G_A           NaN       NaN           NaN  0.000003   \n",
       "72   12_40322047_C_A           NaN       NaN           NaN  0.000003   \n",
       "73   12_40322130_C_A           NaN       NaN           NaN  0.000003   \n",
       "74   12_40322145_A_C           NaN       NaN           NaN  0.000003   \n",
       "75   12_40322322_G_A           NaN       NaN           NaN  0.000003   \n",
       "76   12_40322468_C_A           NaN       NaN           NaN  0.000003   \n",
       "77   12_40323255_A_G           NaN       NaN           NaN  0.000003   \n",
       "78   12_40340380_A_G           NaN       NaN           NaN  0.000012   \n",
       "79   12_40351572_T_A           NaN       NaN           NaN  0.000003   \n",
       "80   12_40354474_A_C           NaN       NaN           NaN  0.000003   \n",
       "81   12_40359344_A_G           NaN       NaN           NaN  0.002956   \n",
       "82   12_40359354_A_G           NaN       NaN           NaN  0.000003   \n",
       "83   12_40363409_A_G           NaN       NaN           NaN  0.000003   \n",
       "84   12_40363541_G_A           NaN       NaN           NaN  0.000003   \n",
       "85   12_40364850_A_G           NaN       NaN           NaN  0.003586   \n",
       "86   12_40367045_G_A           NaN       NaN           NaN  0.000003   \n",
       "87   12_40225159_G_A           NaN       NaN           NaN       NaN   \n",
       "88   12_40235700_A_C           NaN       NaN           NaN       NaN   \n",
       "89   12_40240485_A_C           NaN       NaN           NaN       NaN   \n",
       "90   12_40251361_A_G           NaN       NaN           NaN       NaN   \n",
       "91   12_40251471_G_A           NaN       NaN           NaN       NaN   \n",
       "92   12_40252990_C_A           NaN       NaN           NaN       NaN   \n",
       "93   12_40257334_C_A           NaN       NaN           NaN       NaN   \n",
       "94   12_40263875_A_G           NaN       NaN           NaN       NaN   \n",
       "95   12_40274661_A_G           NaN       NaN           NaN       NaN   \n",
       "96   12_40278154_A_G           NaN       NaN           NaN       NaN   \n",
       "97   12_40284011_G_A           NaN       NaN           NaN       NaN   \n",
       "98   12_40284061_A_G           NaN       NaN           NaN       NaN   \n",
       "99   12_40284113_G_A           NaN       NaN           NaN       NaN   \n",
       "100  12_40284132_I_D           NaN       NaN           NaN       NaN   \n",
       "101  12_40287438_A_G           NaN       NaN           NaN       NaN   \n",
       "102  12_40294866_C_A           NaN       NaN           NaN       NaN   \n",
       "103  12_40294879_G_A           NaN       NaN           NaN       NaN   \n",
       "104  12_40295466_G_A           NaN       NaN           NaN       NaN   \n",
       "105  12_40295567_A_C           NaN       NaN           NaN       NaN   \n",
       "106  12_40299216_C_A           NaN       NaN           NaN       NaN   \n",
       "107  12_40304040_G_C           NaN       NaN           NaN       NaN   \n",
       "108  12_40308481_G_A           NaN       NaN           NaN       NaN   \n",
       "109  12_40308525_G_A           NaN       NaN           NaN       NaN   \n",
       "110  12_40309225_A_C           NaN       NaN           NaN       NaN   \n",
       "111  12_40309226_A_G           NaN       NaN           NaN       NaN   \n",
       "112  12_40310434_C_G           NaN       NaN           NaN       NaN   \n",
       "113  12_40310561_G_A           NaN       NaN           NaN       NaN   \n",
       "114  12_40310635_A_G           NaN       NaN           NaN       NaN   \n",
       "115  12_40314047_G_A           NaN       NaN           NaN       NaN   \n",
       "116  12_40320013_G_A           NaN       NaN           NaN       NaN   \n",
       "117  12_40320040_G_A           NaN       NaN           NaN       NaN   \n",
       "118  12_40321114_A_G           NaN       NaN           NaN       NaN   \n",
       "119  12_40322047_G_A           NaN       NaN           NaN       NaN   \n",
       "120  12_40322131_A_G           NaN       NaN           NaN       NaN   \n",
       "121  12_40323221_I_D           NaN       NaN           NaN       NaN   \n",
       "122  12_40323270_C_A           NaN       NaN           NaN       NaN   \n",
       "123  12_40328362_G_A           NaN       NaN           NaN       NaN   \n",
       "124  12_40335031_G_A           NaN       NaN           NaN       NaN   \n",
       "125  12_40335091_A_G           NaN       NaN           NaN       NaN   \n",
       "126  12_40340361_A_G           NaN       NaN           NaN       NaN   \n",
       "127  12_40340404_A_G           NaN       NaN           NaN       NaN   \n",
       "128  12_40346830_I_D           NaN       NaN           NaN       NaN   \n",
       "129  12_40351585_G_A           NaN       NaN           NaN       NaN   \n",
       "130  12_40359338_A_G           NaN       NaN           NaN       NaN   \n",
       "131  12_40363540_C_A           NaN       NaN           NaN       NaN   \n",
       "\n",
       "           R2_EUR  \n",
       "0             NaN  \n",
       "1    3.029200e-08  \n",
       "2    1.127060e-03  \n",
       "3    1.699050e-06  \n",
       "4    1.214590e-07  \n",
       "5    1.000000e+00  \n",
       "6    1.018600e-04  \n",
       "7    6.576440e-05  \n",
       "8    3.339440e-07  \n",
       "9    3.787680e-03  \n",
       "10            NaN  \n",
       "11   7.151250e-07  \n",
       "12   1.045930e-04  \n",
       "13   8.491320e-07  \n",
       "14   1.825750e-05  \n",
       "15   4.627960e-06  \n",
       "16   1.819250e-07  \n",
       "17   3.945370e-05  \n",
       "18   3.587880e-04  \n",
       "19   6.677650e-06  \n",
       "20   2.247010e-06  \n",
       "21   1.274230e-03  \n",
       "22   3.031130e-08  \n",
       "23   6.073290e-08  \n",
       "24   5.146180e-06  \n",
       "25            NaN  \n",
       "26   2.427070e-07  \n",
       "27   3.030780e-08  \n",
       "28   6.067330e-08  \n",
       "29   3.030600e-08  \n",
       "30   3.030780e-08  \n",
       "31   1.639500e-06  \n",
       "32   6.064530e-08  \n",
       "33   2.949580e-08  \n",
       "34   2.126820e-07  \n",
       "35   1.097200e-06  \n",
       "36   2.121050e-07  \n",
       "37   1.821460e-07  \n",
       "38            NaN  \n",
       "39   6.069080e-08  \n",
       "40   4.550370e-07  \n",
       "41   6.678690e-07  \n",
       "42   9.091280e-08  \n",
       "43   2.726910e-07  \n",
       "44   3.029900e-08  \n",
       "45            NaN  \n",
       "46            NaN  \n",
       "47   7.279530e-07  \n",
       "48   1.515130e-07  \n",
       "49            NaN  \n",
       "50            NaN  \n",
       "51   3.030950e-08  \n",
       "52   3.031390e-07  \n",
       "53   3.031480e-08  \n",
       "54            NaN  \n",
       "55            NaN  \n",
       "56            NaN  \n",
       "57   3.032880e-08  \n",
       "58   3.031480e-08  \n",
       "59   3.030950e-08  \n",
       "60   1.818830e-07  \n",
       "61   2.121790e-07  \n",
       "62   6.065930e-08  \n",
       "63   3.032700e-08  \n",
       "64            NaN  \n",
       "65            NaN  \n",
       "66   9.091280e-08  \n",
       "67            NaN  \n",
       "68            NaN  \n",
       "69   6.070840e-08  \n",
       "70   6.063830e-08  \n",
       "71   2.947880e-08  \n",
       "72   2.425810e-07  \n",
       "73            NaN  \n",
       "74   3.030250e-08  \n",
       "75            NaN  \n",
       "76   6.073640e-08  \n",
       "77   3.036560e-08  \n",
       "78            NaN  \n",
       "79            NaN  \n",
       "80            NaN  \n",
       "81   1.515040e-07  \n",
       "82   9.094950e-08  \n",
       "83            NaN  \n",
       "84   9.101780e-08  \n",
       "85            NaN  \n",
       "86   3.032880e-08  \n",
       "87   3.033050e-08  \n",
       "88   3.034100e-08  \n",
       "89   3.031650e-08  \n",
       "90   3.031130e-08  \n",
       "91   3.029200e-08  \n",
       "92   6.077160e-08  \n",
       "93   3.030780e-08  \n",
       "94   3.029900e-08  \n",
       "95   3.031130e-08  \n",
       "96   6.059630e-08  \n",
       "97   3.033050e-08  \n",
       "98   6.061380e-08  \n",
       "99   6.061030e-08  \n",
       "100  1.821780e-07  \n",
       "101  6.077510e-08  \n",
       "102  3.030600e-08  \n",
       "103  3.030950e-08  \n",
       "104  3.030600e-08  \n",
       "105  6.058930e-08  \n",
       "106  3.030080e-08  \n",
       "107  1.123400e-06  \n",
       "108  8.190180e-07  \n",
       "109  3.032350e-08  \n",
       "110  6.121670e-08  \n",
       "111  6.076100e-08  \n",
       "112  8.794320e-07  \n",
       "113  9.093380e-08  \n",
       "114  3.030430e-08  \n",
       "115  6.064880e-08  \n",
       "116  3.032880e-08  \n",
       "117  3.033580e-08  \n",
       "118  1.217330e-07  \n",
       "119  2.960340e-08  \n",
       "120  3.030080e-08  \n",
       "121  3.029730e-08  \n",
       "122  6.073290e-08  \n",
       "123  3.030430e-08  \n",
       "124  8.797880e-07  \n",
       "125  9.107570e-08  \n",
       "126  3.029900e-08  \n",
       "127  1.820940e-07  \n",
       "128  8.193500e-07  \n",
       "129  3.950720e-07  \n",
       "130  6.060680e-08  \n",
       "131  9.095480e-08  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.DataFrame({\"SNP_B\":\"\"}, index = [0])\n",
    "\n",
    "labels=['AJ', 'CAH', 'CAS', 'EAS', 'EUR']\n",
    "\n",
    "for label in labels:\n",
    "    \n",
    "    ld = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_a419v_r2.ld\", delim_whitespace = True)\n",
    "    ld_red = ld[[\"SNP_B\", \"R2\"]]\n",
    "    ld_red.rename(columns = {\"R2\":f\"R2_{label}\"}, inplace = True)\n",
    "\n",
    "    df = pd.merge(df, ld_red, on = \"SNP_B\", how = \"outer\")\n",
    "\n",
    "df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "276px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
