{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LRRK2 p.A419V - Age of onset analysis "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Project: Multiancestry LRRK2 p.A419V analysis\n",
    "- Version: Python/3.10.12\n",
    "- Created: 05-MAY-2025\n",
    "- Last Update: 12-JUNE-2025"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Description\n",
    "\n",
    "**1. Baseline information**\n",
    "\n",
    "**2. Data preparation**\n",
    "\n",
    "**3. Age of onset statistical analysis**\n",
    "- Adjusted\n",
    "- Unadjusted\n",
    "\n",
    "**4. Meta-analysis with SG-WES-EAS data**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_131/525924512.py:21: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Import necessary packages\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from io import StringIO\n",
    "from firecloud import api as fapi\n",
    "from IPython.core.display import display, HTML\n",
    "import urllib.parse\n",
    "from google.cloud import bigquery\n",
    "import sys as sys\n",
    "import statistics as st\n",
    "\n",
    "# Define function\n",
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R and its packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install rpy2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "* installing *source* package ‘tidyverse’ ...\n",
      "** package ‘tidyverse’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "*** copying figures\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (tidyverse)\n",
      "* installing *source* package ‘data.table’ ...\n",
      "** package ‘data.table’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc 9.4.0\n",
      "zlib 1.2.11 is available ok\n",
      "* checking if R installation supports OpenMP without any extra hints... yes\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c assign.c -o assign.o\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "** libs\n",
      "using C compiler: ‘gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c between.c -o between.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c bmerge.c -o bmerge.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c chmatch.c -o chmatch.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c cj.c -o cj.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c coalesce.c -o coalesce.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dogroups.c -o dogroups.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fastmean.c -o fastmean.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fcast.c -o fcast.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fifelse.c -o fifelse.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fmelt.c -o fmelt.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c forder.c -o forder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frank.c -o frank.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fread.c -o fread.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c freadR.c -o freadR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c froll.c -o froll.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frollR.c -o frollR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frolladaptive.c -o frolladaptive.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fsort.c -o fsort.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwrite.c -o fwrite.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwriteR.c -o fwriteR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c gsumm.c -o gsumm.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c idatetime.c -o idatetime.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ijoin.c -o ijoin.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c init.c -o init.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c inrange.c -o inrange.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nafill.c -o nafill.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c negate.c -o negate.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nqrecreateindices.c -o nqrecreateindices.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c openmp-utils.c -o openmp-utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c programming.c -o programming.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c quickselect.c -o quickselect.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rbindlist.c -o rbindlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c reorder.c -o reorder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c shift.c -o shift.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c snprintf.c -o snprintf.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c subset.c -o subset.o\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c transpose.c -o transpose.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c types.c -o types.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c uniqlist.c -o uniqlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c utils.c -o utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c vecseq.c -o vecseq.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c wrappers.c -o wrappers.o\n",
      "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o data.table.so assign.o between.o bmerge.o chmatch.o cj.o coalesce.o dogroups.o fastmean.o fcast.o fifelse.o fmelt.o forder.o frank.o fread.o freadR.o froll.o frollR.o frolladaptive.o fsort.o fwrite.o fwriteR.o gsumm.o idatetime.o ijoin.o init.o inrange.o nafill.o negate.o nqrecreateindices.o openmp-utils.o programming.o quickselect.o rbindlist.o reorder.o shift.o snprintf.o subset.o transpose.o types.o uniqlist.o utils.o vecseq.o wrappers.o -fopenmp -L/usr/lib/x86_64-linux-gnu -lz -L/usr/lib/R/lib -lR\n",
      "PKG_CFLAGS = -fopenmp\n",
      "PKG_LIBS = -fopenmp -L/usr/lib/x86_64-linux-gnu -lz\n",
      "if [ \"data.table.so\" != \"data_table.so\" ]; then mv data.table.so data_table.so; fi\n",
      "if [ \"\" != \"Windows_NT\" ] && [ `uname -s` = 'Darwin' ]; then install_name_tool -id data_table.so data_table.so; fi\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "installing to /home/jupyter/packages/00LOCK-data.table/00new/data.table/libs\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** checking absolute paths in shared objects and dynamic libraries\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (data.table)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.5\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.5.1     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.4     ✔ tidyr     1.3.1\n",
      "✔ purrr     1.0.4     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Installing package into ‘/home/jupyter/packages’\n",
       "(as ‘lib’ is unspecified)\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/tidyverse_2.0.0.tar.gz'\n",
       "Content type 'application/x-gzip' length 704618 bytes (688 KB)\n",
       "==================================================\n",
       "downloaded 688 KB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/RtmpfesMak/downloaded_packages’\n",
       "Installing package into ‘/home/jupyter/packages’\n",
       "(as ‘lib’ is unspecified)\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/data.table_1.17.4.tar.gz'\n",
       "Content type 'application/x-gzip' length 5839682 bytes (5.6 MB)\n",
       "==================================================\n",
       "downloaded 5.6 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/RtmpfesMak/downloaded_packages’\n",
       "data.table 1.17.4 using 4 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "\n",
       "Attaching package: ‘data.table’\n",
       "\n",
       "The following objects are masked from ‘package:lubridate’:\n",
       "\n",
       "    hour, isoweek, mday, minute, month, quarter, second, wday, week,\n",
       "    yday, year\n",
       "\n",
       "The following objects are masked from ‘package:dplyr’:\n",
       "\n",
       "    between, first, last\n",
       "\n",
       "The following object is masked from ‘package:purrr’:\n",
       "\n",
       "    transpose\n",
       "\n",
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  library ‘/usr/lib/R/site-library’ contains no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "install.packages(\"tidyverse\")\n",
    "install.packages(\"data.table\")\n",
    "install.packages(\"rmeta\")\n",
    "\n",
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(rmeta)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Baseline information "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>mean AAO +- SD</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AAC</td>\n",
       "      <td>59.08 +- 12.49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AFR</td>\n",
       "      <td>57.18 +- 12.98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AJ</td>\n",
       "      <td>62.45 +- 11.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AMR</td>\n",
       "      <td>52.41 +- 13.74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CAH</td>\n",
       "      <td>55.57 +- 13.68</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CAS</td>\n",
       "      <td>52.76 +- 12.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>EAS</td>\n",
       "      <td>53.5 +- 12.88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>EUR</td>\n",
       "      <td>58.94 +- 12.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>FIN</td>\n",
       "      <td>58.38 +- 11.83</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>MDE</td>\n",
       "      <td>53.35 +- 13.38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SAS</td>\n",
       "      <td>52.92 +- 12.86</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   label  mean AAO +- SD\n",
       "0    AAC  59.08 +- 12.49\n",
       "1    AFR  57.18 +- 12.98\n",
       "2     AJ  62.45 +- 11.38\n",
       "3    AMR  52.41 +- 13.74\n",
       "4    CAH  55.57 +- 13.68\n",
       "5    CAS  52.76 +- 12.12\n",
       "6    EAS   53.5 +- 12.88\n",
       "7    EUR   58.94 +- 12.0\n",
       "8    FIN  58.38 +- 11.83\n",
       "9    MDE  53.35 +- 13.38\n",
       "10   SAS  52.92 +- 12.86"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = []\n",
    "\n",
    "labels = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'FIN', 'MDE', 'SAS']\n",
    "for label in labels:\n",
    "    master     = pd.read_csv(f\"/home/jupyter/A419V_release9/master_key_release9_final.csv\")\n",
    "    master_red = master[(master[\"nba_label\"] == label) & (~master[\"age_of_onset\"].isna())]\n",
    "    sd = round(st.stdev(master_red[\"age_of_onset\"]), 2)\n",
    "    mean = round(st.mean(master_red[\"age_of_onset\"]), 2)\n",
    "\n",
    "    df.append({\n",
    "        'label' : label,\n",
    "        'mean AAO +- SD' : str(mean) + ' +- ' + str(sd) \n",
    "    })\n",
    "    \n",
    "pd.DataFrame(df)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data preparation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"/home/jupyter/A419V_release9/extract_snps_raw_cas.txt\", \"w\") as f:\n",
    "    f.write(\"Seq_rs34594498\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"/home/jupyter/A419V_release9/extract_snps_raw.txt\", \"w\") as f:\n",
    "    f.write(\"exm994472\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_remove_related_updated\n",
      "  --extract extract_snps_raw.txt\n",
      "  --filter-cases\n",
      "  --make-bed\n",
      "  --out CAH/CAH_release9_extracted_raw\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1904683 variants loaded from .bim file.\n",
      "982 people (514 males, 468 females) loaded from .fam.\n",
      "954 phenotype values loaded from .fam.\n",
      "--extract: 1 variant remaining.\n",
      "338 people removed due to case/control status (--filter-cases).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 644 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 644 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 0 are controls.\n",
      "--make-bed to CAH/CAH_release9_extracted_raw.bed +\n",
      "CAH/CAH_release9_extracted_raw.bim + CAH/CAH_release9_extracted_raw.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_remove_related_updated\n",
      "  --extract extract_snps_raw.txt\n",
      "  --filter-cases\n",
      "  --make-bed\n",
      "  --out EAS/EAS_release9_extracted_raw\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1730431 variants loaded from .bim file.\n",
      "5646 people (3495 males, 2151 females) loaded from .fam.\n",
      "5571 phenotype values loaded from .fam.\n",
      "--extract: 1 variant remaining.\n",
      "2454 people removed due to case/control status (--filter-cases).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3192 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999373.\n",
      "1 variant and 3192 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 0 are controls.\n",
      "--make-bed to EAS/EAS_release9_extracted_raw.bed +\n",
      "EAS/EAS_release9_extracted_raw.bim + EAS/EAS_release9_extracted_raw.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_remove_related_updated\n",
      "  --extract extract_snps_raw.txt\n",
      "  --filter-cases\n",
      "  --make-bed\n",
      "  --out EUR/EUR_release9_extracted_raw\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1668646 variants loaded from .bim file.\n",
      "34703 people (19239 males, 15464 females) loaded from .fam.\n",
      "20824 phenotype values loaded from .fam.\n",
      "--extract: 1 variant remaining.\n",
      "19371 people removed due to case/control status (--filter-cases).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 15332 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.999283.\n",
      "1 variant and 15332 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 0 are controls.\n",
      "--make-bed to EUR/EUR_release9_extracted_raw.bed +\n",
      "EUR/EUR_release9_extracted_raw.bim + EUR/EUR_release9_extracted_raw.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAH' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_updated \\\n",
    "    --filter-cases \\\n",
    "    --extract extract_snps_raw.txt \\\n",
    "    --make-bed \\\n",
    "    --out ${label}/${label}_release9_extracted_raw\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_remove_related_updated\n",
      "  --extract extract_snps_raw_cas.txt\n",
      "  --filter-cases\n",
      "  --make-bed\n",
      "  --out CAS/CAS_release9_extracted_raw\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1902516 variants loaded from .bim file.\n",
      "1006 people (461 males, 545 females) loaded from .fam.\n",
      "990 phenotype values loaded from .fam.\n",
      "--extract: 1 variant remaining.\n",
      "345 people removed due to case/control status (--filter-cases).\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 661 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 0 are controls.\n",
      "--make-bed to CAS/CAS_release9_extracted_raw.bed +\n",
      "CAS/CAS_release9_extracted_raw.bim + CAS/CAS_release9_extracted_raw.fam ...\n",
      "101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAS')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_updated \\\n",
    "    --filter-cases \\\n",
    "    --extract extract_snps_raw_cas.txt \\\n",
    "    --make-bed \\\n",
    "    --out ${label}/${label}_release9_extracted_raw\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "WORK_DIR = \"/home/jupyter/A419V_release9\"\n",
    "labels = ['CAH', 'CAS', 'EAS', 'EUR']\n",
    "\n",
    "for label in labels:\n",
    "    \n",
    "    # Rename the SNPID in bim file to be in the format of chr_pos_ref_alt\n",
    "    bim = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_release9_extracted_raw.bim\", sep = \"\\t\", names = [\"chr\", \"rsid\", \"pos\", \"bp\", \"a1\", \"a2\"])\n",
    "    bim['chr'] = 'chr' + bim['chr'].astype(str)\n",
    "    bim['bp'] = bim['bp'].astype(str)\n",
    "    bim['rsid'] = bim['chr'].str.cat(bim['bp'], sep = \"_\")\n",
    "    bim['rsid'] = bim['rsid'].str.cat(bim['a2'], sep = \"_\")\n",
    "    bim['rsid'] = bim['rsid'].str.cat(bim['a1'], sep = \"_\")\n",
    "    bim.to_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_release9_extracted_raw.bim\", sep = \"\\t\", index = False, header = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_extracted_raw\n",
      "  --out CAH/CAH_release9_extracted_raw\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "644 people (369 males, 275 females) loaded from .fam.\n",
      "644 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 644 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 644 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 0 are controls.\n",
      "--recode A to CAH/CAH_release9_extracted_raw.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_extracted_raw\n",
      "  --out EAS/EAS_release9_extracted_raw\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "3192 people (1767 males, 1425 females) loaded from .fam.\n",
      "3192 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3192 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999373.\n",
      "1 variant and 3192 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 0 are controls.\n",
      "--recode A to EAS/EAS_release9_extracted_raw.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_extracted_raw\n",
      "  --out EUR/EUR_release9_extracted_raw\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "15332 people (9549 males, 5783 females) loaded from .fam.\n",
      "15332 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 15332 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999283.\n",
      "1 variant and 15332 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 0 are controls.\n",
      "--recode A to EUR/EUR_release9_extracted_raw.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_release9_extracted_raw.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_extracted_raw\n",
      "  --out CAS/CAS_release9_extracted_raw\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "661 people (301 males, 360 females) loaded from .fam.\n",
      "661 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 661 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 0 are controls.\n",
      "--recode A to CAS/CAS_release9_extracted_raw.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAH' 'EAS' 'EUR' 'CAS')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_extracted_raw \\\n",
    "    --recode A \\\n",
    "    --out ${label}/${label}_release9_extracted_raw\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Covariate file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['IID', 'study', 'nba', 'wgs', 'clinical_exome',\n",
       "       'extended_clinical_data', 'GDPR', 'nba_prune_reason', 'nba_related',\n",
       "       'nba_label', 'nba_GP2sampleID_r7', 'wgs_prune_reason', 'wgs_flag',\n",
       "       'wgs_label', 'wgs_GP2ID_r8', 'study_arm', 'study_type', 'diagnosis',\n",
       "       'baseline_GP2_phenotype_for_qc', 'baseline_GP2_phenotype',\n",
       "       'biological_sex_for_qc', 'age_at_sample_collection', 'age_of_onset',\n",
       "       'age_at_diagnosis', 'age_at_death', 'age_at_last_follow_up',\n",
       "       'race_for_qc', 'family_history_for_qc', 'region_for_qc', 'manifest_id',\n",
       "       'Genotyping_site', 'sample_type'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels=['CAH', 'CAS', 'EAS', 'EUR', 'AJ']\n",
    "for label in labels:\n",
    "    \n",
    "    cov = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_covar.txt\", sep = \"\\t\")\n",
    "    master = pd.read_csv(f\"/home/jupyter/A419V_release9/master_key_release9_final.csv\")\n",
    "    master.rename(columns = {\"GP2ID\":\"IID\"}, inplace = True)\n",
    "\n",
    "    AAO = master[[\"IID\", \"age_of_onset\", \"region_for_qc\"]]\n",
    "\n",
    "    cov_AAO = pd.merge(cov, AAO, how = \"left\", on = \"IID\")\n",
    "    cov_AAO_cases = cov_AAO[(cov_AAO[\"PHENO\"] == 2) & (~cov_AAO[\"age_of_onset\"].isna())]\n",
    "    cov_AAO_cases[[\"FID\", \"IID\", \"age_of_onset\"]].to_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_covar_AAO.txt\", sep = \"\\t\", header = True, index = False)\n",
    "    cov_AAO_cases[[\"FID\", \"IID\", \"SEX\", \"age_of_onset\", \"region_for_qc\", \"PC1\", \"PC2\", \"PC3\", \"PC4\", \"PC5\"]].to_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_covar_AAO_full.txt\", sep = \"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Age of onset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('AJ' 'CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "    \n",
    "    cat ${label}/${label}_covar_AAO.txt >> all_covar_AAO.txt\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15305 all_covar_AAO.txt\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "wc -l all_covar_AAO.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Linear model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### No adjustment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --glm allow-no-covars cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out CAH/CAH_AAO_unadjusted\n",
      "  --pheno CAH/CAH_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:32:19 2025\n",
      "52216 MiB RAM detected, ~50558 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "644 samples (275 females, 369 males; 644 founders) loaded from\n",
      "CAH/CAH_release9_extracted_raw.fam.\n",
      "1 variant loaded from CAH/CAH_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (408 values).\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to CAH/CAH_AAO_unadjusted.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:32:20 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --glm allow-no-covars cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out CAS/CAS_AAO_unadjusted\n",
      "  --pheno CAS/CAS_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:32:20 2025\n",
      "52216 MiB RAM detected, ~50557 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "661 samples (360 females, 301 males; 661 founders) loaded from\n",
      "CAS/CAS_release9_extracted_raw.fam.\n",
      "1 variant loaded from CAS/CAS_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (459 values).\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to CAS/CAS_AAO_unadjusted.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:32:20 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --glm allow-no-covars cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out EAS/EAS_AAO_unadjusted\n",
      "  --pheno EAS/EAS_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:32:20 2025\n",
      "52216 MiB RAM detected, ~50557 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "3192 samples (1425 females, 1767 males; 3192 founders) loaded from\n",
      "EAS/EAS_release9_extracted_raw.fam.\n",
      "1 variant loaded from EAS/EAS_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (2366 values).\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to EAS/EAS_AAO_unadjusted.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:32:20 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --glm allow-no-covars cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out EUR/EUR_AAO_unadjusted\n",
      "  --pheno EUR/EUR_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:32:20 2025\n",
      "52216 MiB RAM detected, ~50557 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "15332 samples (5783 females, 9549 males; 15332 founders) loaded from\n",
      "EUR/EUR_release9_extracted_raw.fam.\n",
      "1 variant loaded from EUR/EUR_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (10726 values).\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to EUR/EUR_AAO_unadjusted.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:32:20 2025\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink2 \\\n",
    "    --bfile ${label}/${label}_release9_extracted_raw \\\n",
    "    --pheno ${label}/${label}_covar_AAO.txt \\\n",
    "    --pheno-name age_of_onset \\\n",
    "    --glm allow-no-covars cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \\\n",
    "    --ci 0.95 \\\n",
    "    --out ${label}/${label}_AAO_unadjusted\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_84/4179718931.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/4179718931.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/4179718931.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/4179718931.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/4179718931.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/4179718931.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/4179718931.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/4179718931.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>A1_CT</th>\n",
       "      <th>REF_CT</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>T_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAH</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>816</td>\n",
       "      <td>3</td>\n",
       "      <td>813</td>\n",
       "      <td>0.003676</td>\n",
       "      <td>ADD</td>\n",
       "      <td>408</td>\n",
       "      <td>-6.59412</td>\n",
       "      <td>8.09728</td>\n",
       "      <td>-22.46450</td>\n",
       "      <td>9.276260</td>\n",
       "      <td>-0.814362</td>\n",
       "      <td>0.415915</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAS</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>918</td>\n",
       "      <td>10</td>\n",
       "      <td>908</td>\n",
       "      <td>0.010893</td>\n",
       "      <td>ADD</td>\n",
       "      <td>459</td>\n",
       "      <td>-7.99510</td>\n",
       "      <td>3.75820</td>\n",
       "      <td>-15.36100</td>\n",
       "      <td>-0.629163</td>\n",
       "      <td>-2.127370</td>\n",
       "      <td>0.033924</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EAS</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4730</td>\n",
       "      <td>74</td>\n",
       "      <td>4656</td>\n",
       "      <td>0.015645</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2365</td>\n",
       "      <td>-2.97755</td>\n",
       "      <td>1.47309</td>\n",
       "      <td>-5.86476</td>\n",
       "      <td>-0.090347</td>\n",
       "      <td>-2.021300</td>\n",
       "      <td>0.043362</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EUR</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>21436</td>\n",
       "      <td>14</td>\n",
       "      <td>21422</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10718</td>\n",
       "      <td>-2.40034</td>\n",
       "      <td>3.00913</td>\n",
       "      <td>-8.29813</td>\n",
       "      <td>3.497460</td>\n",
       "      <td>-0.797684</td>\n",
       "      <td>0.425072</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  label                  ID REF ALT  ALLELE_CT  A1_CT  REF_CT   A1_FREQ TEST  \\\n",
       "0   CAH  chr12_40252984_G_A   G   A        816      3     813  0.003676  ADD   \n",
       "0   CAS  chr12_40252984_G_A   G   A        918     10     908  0.010893  ADD   \n",
       "0   EAS  chr12_40252984_G_A   G   A       4730     74    4656  0.015645  ADD   \n",
       "0   EUR  chr12_40252984_G_A   G   A      21436     14   21422  0.000653  ADD   \n",
       "\n",
       "   OBS_CT     BETA       SE       L95       U95    T_STAT         P ERRCODE  \n",
       "0     408 -6.59412  8.09728 -22.46450  9.276260 -0.814362  0.415915       .  \n",
       "0     459 -7.99510  3.75820 -15.36100 -0.629163 -2.127370  0.033924       .  \n",
       "0    2365 -2.97755  1.47309  -5.86476 -0.090347 -2.021300  0.043362       .  \n",
       "0   10718 -2.40034  3.00913  -8.29813  3.497460 -0.797684  0.425072       .  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels = ['CAH', 'CAS', 'EAS', 'EUR']\n",
    "df = pd.DataFrame()\n",
    "\n",
    "for label in labels:\n",
    "\n",
    "    lm          = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_AAO_unadjusted.age_of_onset.glm.linear\", delim_whitespace = True)\n",
    "    lm_a419     = lm[lm[\"ID\"] == \"chr12_40252984_G_A\"]\n",
    "    lm_a419_red = lm_a419[[\"ID\", \"REF\", \"ALT\", \"A1_CT\", \"ALLELE_CT\", \"A1_FREQ\", \"TEST\", \"OBS_CT\", \"BETA\", \"SE\", \"L95\", \"U95\", \"T_STAT\", \"P\", \"ERRCODE\"]]\n",
    "    \n",
    "    lm_a419_red[\"label\"]  = label\n",
    "    lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
    "    lm_a419_red           = lm_a419_red[[\"label\", \"ID\", \"REF\", \"ALT\", \"ALLELE_CT\", \"A1_CT\", \"REF_CT\", \"A1_FREQ\", \"TEST\", \"OBS_CT\", \"BETA\", \"SE\", \"L95\", \"U95\", \"T_STAT\", \"P\", \"ERRCODE\"]]\n",
    "    \n",
    "    df = pd.concat([df, lm_a419_red])\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Add on info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_extracted_raw\n",
      "  --out CAH/CAH_AAO_unadjusted\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "644 people (369 males, 275 females) loaded from .fam.\n",
      "644 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 644 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 644 people pass filters and QC.\n",
      "Among remaining phenotypes, 644 are cases and 0 are controls.\n",
      "--recode A to CAH/CAH_AAO_unadjusted.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_extracted_raw\n",
      "  --out CAS/CAS_AAO_unadjusted\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "661 people (301 males, 360 females) loaded from .fam.\n",
      "661 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 661 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "1 variant and 661 people pass filters and QC.\n",
      "Among remaining phenotypes, 661 are cases and 0 are controls.\n",
      "--recode A to CAS/CAS_AAO_unadjusted.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_extracted_raw\n",
      "  --out EAS/EAS_AAO_unadjusted\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "3192 people (1767 males, 1425 females) loaded from .fam.\n",
      "3192 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 3192 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999373.\n",
      "1 variant and 3192 people pass filters and QC.\n",
      "Among remaining phenotypes, 3192 are cases and 0 are controls.\n",
      "--recode A to EAS/EAS_AAO_unadjusted.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "PLINK v1.90b6.9 64-bit (4 Mar 2019)            www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2019 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_AAO_unadjusted.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_extracted_raw\n",
      "  --out EUR/EUR_AAO_unadjusted\n",
      "  --recode A\n",
      "\n",
      "52216 MB RAM detected; reserving 26108 MB for main workspace.\n",
      "1 variant loaded from .bim file.\n",
      "15332 people (9549 males, 5783 females) loaded from .fam.\n",
      "15332 phenotype values loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 15332 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.999283.\n",
      "1 variant and 15332 people pass filters and QC.\n",
      "Among remaining phenotypes, 15332 are cases and 0 are controls.\n",
      "--recode A to EUR/EUR_AAO_unadjusted.raw ... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_extracted_raw \\\n",
    "    --recode A \\\n",
    "    --out ${label}/${label}_AAO_unadjusted\n",
    "    \n",
    "done "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Hom Ref Cases</th>\n",
       "      <th>Missing Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier Freq in Cases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAS</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>449</td>\n",
       "      <td>0</td>\n",
       "      <td>459</td>\n",
       "      <td>0.010893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EAS</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>72</td>\n",
       "      <td>2292</td>\n",
       "      <td>1</td>\n",
       "      <td>2366</td>\n",
       "      <td>0.015645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EUR</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>10705</td>\n",
       "      <td>8</td>\n",
       "      <td>10726</td>\n",
       "      <td>0.000653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAH</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>405</td>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>0.003676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ancestry               Variant  Hom Cases  Het Cases  Hom Ref Cases  \\\n",
       "0      CAS  chr12_40252984_G_A_A          0         10            449   \n",
       "1      EAS  chr12_40252984_G_A_A          1         72           2292   \n",
       "2      EUR  chr12_40252984_G_A_A          1         12          10705   \n",
       "3      CAH  chr12_40252984_G_A_A          0          3            405   \n",
       "\n",
       "   Missing Cases  Total Cases  Carrier Freq in Cases  \n",
       "0              0          459               0.010893  \n",
       "1              1         2366               0.015645  \n",
       "2              8        10726               0.000653  \n",
       "3              0          408               0.003676  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels = ['CAS', 'EAS', 'EUR', 'CAH']\n",
    "variant = \"chr12_40252984_G_A_A\"\n",
    "results = []  \n",
    "\n",
    "for label in labels:\n",
    "    \n",
    "    recode = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_AAO_unadjusted.raw\", \n",
    "                         delim_whitespace=True)\n",
    "    \n",
    "    cov = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_covar_AAO.txt\",\n",
    "                      sep= \"\\t\")\n",
    "    \n",
    "    cov_red = cov[[\"IID\", \"age_of_onset\"]]\n",
    "    \n",
    "    merge = pd.merge(recode, cov_red, on = \"IID\", how = \"left\")\n",
    "    \n",
    "    merge = merge[~merge[\"age_of_onset\"].isna()]\n",
    "    \n",
    "    cases_data = merge[merge['PHENOTYPE'] == 2]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    \n",
    "    # Cases\n",
    "    hom_cases = (cases_data[variant] == 2).sum()\n",
    "    het_cases = (cases_data[variant] == 1).sum()\n",
    "    hom_ref_cases = (cases_data[variant] == 0).sum()\n",
    "    missing_cases = total_cases - (hom_cases + het_cases + hom_ref_cases)\n",
    "    freq_cases = (2 * hom_cases + het_cases) / (2 * (total_cases - missing_cases)) if (total_cases - missing_cases) > 0 else None\n",
    "\n",
    "    # Collect results\n",
    "    results.append({\n",
    "        'Ancestry': label,\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Hom Ref Cases': hom_ref_cases,\n",
    "        'Missing Cases': missing_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier Freq in Cases': freq_cases,\n",
    "    })\n",
    "\n",
    "# Convert to DataFrame\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Adjusted"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Adjusted by sex and PCs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAH/CAH_AAO_adjusted_sex_pc.log.\n",
      "Options in effect:\n",
      "  --bfile CAH/CAH_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --covar CAH/CAH_covar_AAO_full.txt\n",
      "  --covar-name SEX,PC1,PC2,PC3,PC4,PC5\n",
      "  --glm cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out CAH/CAH_AAO_adjusted_sex_pc\n",
      "  --pheno CAH/CAH_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:33:02 2025\n",
      "52216 MiB RAM detected, ~50549 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "644 samples (275 females, 369 males; 644 founders) loaded from\n",
      "CAH/CAH_release9_extracted_raw.fam.\n",
      "1 variant loaded from CAH/CAH_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (408 values).\n",
      "6 covariates loaded from CAH/CAH_covar_AAO_full.txt.\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to CAH/CAH_AAO_adjusted_sex_pc.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:33:02 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to CAS/CAS_AAO_adjusted_sex_pc.log.\n",
      "Options in effect:\n",
      "  --bfile CAS/CAS_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --covar CAS/CAS_covar_AAO_full.txt\n",
      "  --covar-name SEX,PC1,PC2,PC3,PC4,PC5\n",
      "  --glm cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out CAS/CAS_AAO_adjusted_sex_pc\n",
      "  --pheno CAS/CAS_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:33:02 2025\n",
      "52216 MiB RAM detected, ~50548 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "661 samples (360 females, 301 males; 661 founders) loaded from\n",
      "CAS/CAS_release9_extracted_raw.fam.\n",
      "1 variant loaded from CAS/CAS_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (459 values).\n",
      "6 covariates loaded from CAS/CAS_covar_AAO_full.txt.\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to CAS/CAS_AAO_adjusted_sex_pc.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:33:02 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EAS/EAS_AAO_adjusted_sex_pc.log.\n",
      "Options in effect:\n",
      "  --bfile EAS/EAS_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --covar EAS/EAS_covar_AAO_full.txt\n",
      "  --covar-name SEX,PC1,PC2,PC3,PC4,PC5\n",
      "  --glm cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out EAS/EAS_AAO_adjusted_sex_pc\n",
      "  --pheno EAS/EAS_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:33:02 2025\n",
      "52216 MiB RAM detected, ~50548 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "3192 samples (1425 females, 1767 males; 3192 founders) loaded from\n",
      "EAS/EAS_release9_extracted_raw.fam.\n",
      "1 variant loaded from EAS/EAS_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (2366 values).\n",
      "6 covariates loaded from EAS/EAS_covar_AAO_full.txt.\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to EAS/EAS_AAO_adjusted_sex_pc.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:33:02 2025\n",
      "PLINK v2.0.0-a.6.9LM 64-bit Intel (29 Jan 2025)    cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2025 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to EUR/EUR_AAO_adjusted_sex_pc.log.\n",
      "Options in effect:\n",
      "  --bfile EUR/EUR_release9_extracted_raw\n",
      "  --ci 0.95\n",
      "  --covar EUR/EUR_covar_AAO_full.txt\n",
      "  --covar-name SEX,PC1,PC2,PC3,PC4,PC5\n",
      "  --glm cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err\n",
      "  --out EUR/EUR_AAO_adjusted_sex_pc\n",
      "  --pheno EUR/EUR_covar_AAO.txt\n",
      "  --pheno-name age_of_onset\n",
      "\n",
      "Start time: Mon Jun  9 13:33:02 2025\n",
      "52216 MiB RAM detected, ~50548 available; reserving 26108 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "15332 samples (5783 females, 9549 males; 15332 founders) loaded from\n",
      "EUR/EUR_release9_extracted_raw.fam.\n",
      "1 variant loaded from EUR/EUR_release9_extracted_raw.bim.\n",
      "1 quantitative phenotype loaded (10726 values).\n",
      "6 covariates loaded from EUR/EUR_covar_AAO_full.txt.\n",
      "Calculating allele frequencies... done.\n",
      "--glm linear regression on phenotype 'age_of_onset': done.\n",
      "Results written to EUR/EUR_AAO_adjusted_sex_pc.age_of_onset.glm.linear .\n",
      "End time: Mon Jun  9 13:33:02 2025\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('CAH' 'CAS' 'EAS' 'EUR')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink2 \\\n",
    "    --bfile ${label}/${label}_release9_extracted_raw \\\n",
    "    --pheno ${label}/${label}_covar_AAO.txt \\\n",
    "    --pheno-name age_of_onset \\\n",
    "    --covar ${label}/${label}_covar_AAO_full.txt \\\n",
    "    --covar-name SEX,PC1,PC2,PC3,PC4,PC5 \\\n",
    "    --glm cols=+a1freq,+a1freqcc,+a1count,+totallele,+a1countcc,+totallelecc,+err \\\n",
    "    --ci 0.95 \\\n",
    "    --out ${label}/${label}_AAO_adjusted_sex_pc\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_84/2916850425.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/2916850425.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/2916850425.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/2916850425.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/2916850425.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/2916850425.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
      "/tmp/ipykernel_84/2916850425.py:10: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"label\"]  = label\n",
      "/tmp/ipykernel_84/2916850425.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>label</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>ALLELE_CT</th>\n",
       "      <th>A1_CT</th>\n",
       "      <th>REF_CT</th>\n",
       "      <th>A1_FREQ</th>\n",
       "      <th>TEST</th>\n",
       "      <th>OBS_CT</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>L95</th>\n",
       "      <th>U95</th>\n",
       "      <th>T_STAT</th>\n",
       "      <th>P</th>\n",
       "      <th>ERRCODE</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAH</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>816</td>\n",
       "      <td>3</td>\n",
       "      <td>813</td>\n",
       "      <td>0.003676</td>\n",
       "      <td>ADD</td>\n",
       "      <td>408</td>\n",
       "      <td>-10.391000</td>\n",
       "      <td>8.17257</td>\n",
       "      <td>-26.40890</td>\n",
       "      <td>5.626970</td>\n",
       "      <td>-1.271440</td>\n",
       "      <td>0.204309</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAS</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>918</td>\n",
       "      <td>10</td>\n",
       "      <td>908</td>\n",
       "      <td>0.010893</td>\n",
       "      <td>ADD</td>\n",
       "      <td>459</td>\n",
       "      <td>-9.497860</td>\n",
       "      <td>3.76428</td>\n",
       "      <td>-16.87570</td>\n",
       "      <td>-2.120000</td>\n",
       "      <td>-2.523150</td>\n",
       "      <td>0.011973</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EAS</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>4730</td>\n",
       "      <td>74</td>\n",
       "      <td>4656</td>\n",
       "      <td>0.015645</td>\n",
       "      <td>ADD</td>\n",
       "      <td>2365</td>\n",
       "      <td>-2.917780</td>\n",
       "      <td>1.46359</td>\n",
       "      <td>-5.78637</td>\n",
       "      <td>-0.049189</td>\n",
       "      <td>-1.993570</td>\n",
       "      <td>0.046314</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>EUR</td>\n",
       "      <td>chr12_40252984_G_A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>21436</td>\n",
       "      <td>14</td>\n",
       "      <td>21422</td>\n",
       "      <td>0.000653</td>\n",
       "      <td>ADD</td>\n",
       "      <td>10718</td>\n",
       "      <td>-0.355029</td>\n",
       "      <td>3.03352</td>\n",
       "      <td>-6.30062</td>\n",
       "      <td>5.590570</td>\n",
       "      <td>-0.117035</td>\n",
       "      <td>0.906834</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  label                  ID REF ALT  ALLELE_CT  A1_CT  REF_CT   A1_FREQ TEST  \\\n",
       "0   CAH  chr12_40252984_G_A   G   A        816      3     813  0.003676  ADD   \n",
       "0   CAS  chr12_40252984_G_A   G   A        918     10     908  0.010893  ADD   \n",
       "0   EAS  chr12_40252984_G_A   G   A       4730     74    4656  0.015645  ADD   \n",
       "0   EUR  chr12_40252984_G_A   G   A      21436     14   21422  0.000653  ADD   \n",
       "\n",
       "   OBS_CT       BETA       SE       L95       U95    T_STAT         P ERRCODE  \n",
       "0     408 -10.391000  8.17257 -26.40890  5.626970 -1.271440  0.204309       .  \n",
       "0     459  -9.497860  3.76428 -16.87570 -2.120000 -2.523150  0.011973       .  \n",
       "0    2365  -2.917780  1.46359  -5.78637 -0.049189 -1.993570  0.046314       .  \n",
       "0   10718  -0.355029  3.03352  -6.30062  5.590570 -0.117035  0.906834       .  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels = ['CAH', 'CAS', 'EAS', 'EUR']\n",
    "df = pd.DataFrame()\n",
    "\n",
    "for label in labels:\n",
    "\n",
    "    lm          = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_AAO_adjusted_sex_pc.age_of_onset.glm.linear\", delim_whitespace = True)\n",
    "    lm_a419     = lm[lm[\"ID\"] == \"chr12_40252984_G_A\"]\n",
    "    lm_a419_red = lm_a419[[\"ID\", \"REF\", \"ALT\", \"A1_CT\", \"ALLELE_CT\", \"A1_FREQ\", \"TEST\", \"OBS_CT\", \"BETA\", \"SE\", \"L95\", \"U95\", \"T_STAT\", \"P\", \"ERRCODE\"]]\n",
    "    \n",
    "    lm_a419_red[\"label\"]  = label\n",
    "    lm_a419_red[\"REF_CT\"] = lm_a419_red[\"ALLELE_CT\"] - lm_a419_red[\"A1_CT\"]\n",
    "    lm_a419_red           = lm_a419_red[[\"label\", \"ID\", \"REF\", \"ALT\", \"ALLELE_CT\", \"A1_CT\", \"REF_CT\", \"A1_FREQ\", \"TEST\", \"OBS_CT\", \"BETA\", \"SE\", \"L95\", \"U95\", \"T_STAT\", \"P\", \"ERRCODE\"]]\n",
    "    \n",
    "    df = pd.concat([df, lm_a419_red])\n",
    "\n",
    "df[df[\"TEST\"] == \"ADD\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Ancestry</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Hom Cases</th>\n",
       "      <th>Het Cases</th>\n",
       "      <th>Hom Ref Cases</th>\n",
       "      <th>Missing Cases</th>\n",
       "      <th>Total Cases</th>\n",
       "      <th>Carrier Freq in Cases</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CAS</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>449</td>\n",
       "      <td>0</td>\n",
       "      <td>459</td>\n",
       "      <td>0.010893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>EAS</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>72</td>\n",
       "      <td>2292</td>\n",
       "      <td>1</td>\n",
       "      <td>2366</td>\n",
       "      <td>0.015645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EUR</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>1</td>\n",
       "      <td>12</td>\n",
       "      <td>10705</td>\n",
       "      <td>8</td>\n",
       "      <td>10726</td>\n",
       "      <td>0.000653</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAH</td>\n",
       "      <td>chr12_40252984_G_A_A</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>405</td>\n",
       "      <td>0</td>\n",
       "      <td>408</td>\n",
       "      <td>0.003676</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Ancestry               Variant  Hom Cases  Het Cases  Hom Ref Cases  \\\n",
       "0      CAS  chr12_40252984_G_A_A          0         10            449   \n",
       "1      EAS  chr12_40252984_G_A_A          1         72           2292   \n",
       "2      EUR  chr12_40252984_G_A_A          1         12          10705   \n",
       "3      CAH  chr12_40252984_G_A_A          0          3            405   \n",
       "\n",
       "   Missing Cases  Total Cases  Carrier Freq in Cases  \n",
       "0              0          459               0.010893  \n",
       "1              1         2366               0.015645  \n",
       "2              8        10726               0.000653  \n",
       "3              0          408               0.003676  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "labels = ['CAS', 'EAS', 'EUR', 'CAH']\n",
    "variant = \"chr12_40252984_G_A_A\"\n",
    "results = []  \n",
    "\n",
    "for label in labels:\n",
    "    \n",
    "    recode = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_AAO_unadjusted.raw\", \n",
    "                         delim_whitespace=True)\n",
    "    \n",
    "    cov = pd.read_csv(f\"/home/jupyter/A419V_release9/{label}/{label}_covar_AAO_full.txt\",\n",
    "                      sep= \"\\t\")\n",
    "    \n",
    "    cov_red = cov[[\"IID\", \"age_of_onset\"]]\n",
    "    \n",
    "    merge = pd.merge(recode, cov_red, on = \"IID\", how = \"left\")\n",
    "    \n",
    "    merge = merge[~merge[\"age_of_onset\"].isna()]\n",
    "    \n",
    "    cases_data = merge[merge['PHENOTYPE'] == 2]\n",
    "    total_cases = cases_data.shape[0]\n",
    "    \n",
    "    # Cases\n",
    "    hom_cases = (cases_data[variant] == 2).sum()\n",
    "    het_cases = (cases_data[variant] == 1).sum()\n",
    "    hom_ref_cases = (cases_data[variant] == 0).sum()\n",
    "    missing_cases = total_cases - (hom_cases + het_cases + hom_ref_cases)\n",
    "    freq_cases = (2 * hom_cases + het_cases) / (2 * (total_cases - missing_cases)) if (total_cases - missing_cases) > 0 else None\n",
    "\n",
    "    # Collect results\n",
    "    results.append({\n",
    "        'Ancestry': label,\n",
    "        'Variant': variant,\n",
    "        'Hom Cases': hom_cases,\n",
    "        'Het Cases': het_cases,\n",
    "        'Hom Ref Cases': hom_ref_cases,\n",
    "        'Missing Cases': missing_cases,\n",
    "        'Total Cases': total_cases,\n",
    "        'Carrier Freq in Cases': freq_cases,\n",
    "    })\n",
    "\n",
    "# Convert to DataFrame\n",
    "df_results = pd.DataFrame(results)\n",
    "df_results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Meta-analysis GP2-EAS & SG-WES-EAS data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Create data frame contain lm outcome of both datasets\n",
    "sumstats <- data.frame(\n",
    "  COHORT = c(\"GP2-EAS\", \"SG-EAS-WES\"),\n",
    "  BETA = c(-2.92, -0.89),\n",
    "  SE = c(1.46, 1.01),\n",
    "  P = c(0.046, 0.38)\n",
    ")\n",
    "\n",
    "# Meta analyse both dataset\n",
    "met <- meta.summaries(d = sumstats$BETA, se = sumstats$SE, method = c(\"fixed\"), logscale = F, names = sumstats$COHORT) # logscale option set for OR.\n",
    "print(met$summary)\n",
    "print(met$se.summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R\n",
    "\n",
    "# Add on the new meta-analysed results\n",
    "dat <- rbind(sumstats, data.frame(COHORT = \"META\", BETA = -1.55, SE = 0.83, P = 0.063))\n",
    "level_order <- c('META', 'SG-EAS-WES', 'GP2-EAS')\n",
    "\n",
    "# Convert the value as numeric\n",
    "dat$BETA <- as.numeric(dat$BETA)\n",
    "dat$SE <- as.numeric(dat$SE)\n",
    "\n",
    "# Create the forest plot\n",
    "resilience_plot <- ggplot(data = dat, aes(x = factor(COHORT, levels = level_order), y = BETA))  +\n",
    "  geom_pointrange(\n",
    "    aes(ymin = BETA - SE,\n",
    "        ymax = BETA + SE),\n",
    "    cex = 0.7\n",
    "  ) +\n",
    "  geom_hline(\n",
    "    yintercept = 0,\n",
    "    linetype = 2\n",
    "  ) +\n",
    "  theme(plot.title = element_text(size = 20, face = \"bold\"),\n",
    "        axis.text.y = element_text(size = 8, face = 'bold'),\n",
    "        axis.ticks.y = element_blank(),\n",
    "        axis.text.x = element_text(face = \"bold\"),\n",
    "        axis.title = element_text(size = 16, face = \"bold\"),\n",
    "        legend.position = \"none\"\n",
    "  ) +\n",
    "  scale_y_continuous(expand = expansion(mult = c(0.1, 0.1)))  + # Allows room for extremes\n",
    "  xlab(' ') +\n",
    "  ylab(\"Beta coefficient\") +\n",
    "  coord_flip() +\n",
    "  theme_minimal() +\n",
    "  ggtitle(\"Forest plot\") +\n",
    "  theme(plot.title = element_text(hjust = 0.5))\n",
    "\n",
    "# Save out plot\n",
    "ggsave(\"Forestplot.jpg\", plot = resilience_plot,\n",
    "       width = 8, height = 5,\n",
    "       units = \"in\",\n",
    "       dpi = 500)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "276px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
