{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Project: Multiancestry LRRK2 p.A419V analysis\n",
    "- Version: Python/3.10.12\n",
    "- Created: 05-MAY-2025\n",
    "- Last Update: 12-JUNE-2025"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Notebook Overview"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- Remove related individual\n",
    "- Files preparation\n",
    "- Baseline differences in sex and fhx\n",
    "- HWE\n",
    "- Allele frequency"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Getting started"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load python libraries"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_130/3484000039.py:7: DeprecationWarning: Importing display from IPython.core.display is deprecated since IPython 7.14, please import from IPython display\n",
      "  from IPython.core.display import display, HTML\n"
     ]
    }
   ],
   "source": [
    "# Import necessary packages\n",
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from io import StringIO\n",
    "from firecloud import api as fapi\n",
    "from IPython.core.display import display, HTML\n",
    "import urllib.parse\n",
    "from google.cloud import bigquery\n",
    "import sys as sys\n",
    "\n",
    "# Define function\n",
    "# Utility routine for printing a shell command before executing it\n",
    "def shell_do(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    !$command\n",
    "    \n",
    "def shell_return(command):\n",
    "    print(f'Executing: {command}', file=sys.stderr)\n",
    "    output = !$command\n",
    "    return '\\n'.join(output)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Install R and load packages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "pip install rpy2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "* installing *source* package ‘tidyverse’ ...\n",
      "** package ‘tidyverse’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "*** copying figures\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (tidyverse)\n",
      "* installing *source* package ‘data.table’ ...\n",
      "** package ‘data.table’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc 9.4.0\n",
      "zlib 1.2.11 is available ok\n",
      "* checking if R installation supports OpenMP without any extra hints... yes\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c assign.c -o assign.o\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "** libs\n",
      "using C compiler: ‘gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0’\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c between.c -o between.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c bmerge.c -o bmerge.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c chmatch.c -o chmatch.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c cj.c -o cj.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c coalesce.c -o coalesce.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c dogroups.c -o dogroups.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fastmean.c -o fastmean.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fcast.c -o fcast.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fifelse.c -o fifelse.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fmelt.c -o fmelt.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c forder.c -o forder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frank.c -o frank.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fread.c -o fread.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c freadR.c -o freadR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c froll.c -o froll.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frollR.c -o frollR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c frolladaptive.c -o frolladaptive.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fsort.c -o fsort.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwrite.c -o fwrite.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c fwriteR.c -o fwriteR.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c gsumm.c -o gsumm.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c idatetime.c -o idatetime.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c ijoin.c -o ijoin.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c init.c -o init.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c inrange.c -o inrange.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nafill.c -o nafill.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c negate.c -o negate.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c nqrecreateindices.c -o nqrecreateindices.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c openmp-utils.c -o openmp-utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c programming.c -o programming.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c quickselect.c -o quickselect.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c rbindlist.c -o rbindlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c reorder.c -o reorder.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c shift.c -o shift.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c snprintf.c -o snprintf.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c subset.c -o subset.o\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c transpose.c -o transpose.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c types.c -o types.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c uniqlist.c -o uniqlist.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c utils.c -o utils.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c vecseq.c -o vecseq.o\n",
      "gcc -I\"/usr/share/R/include\" -DNDEBUG      -fopenmp  -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-EpRONj/r-base-4.4.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c wrappers.c -o wrappers.o\n",
      "gcc -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o data.table.so assign.o between.o bmerge.o chmatch.o cj.o coalesce.o dogroups.o fastmean.o fcast.o fifelse.o fmelt.o forder.o frank.o fread.o freadR.o froll.o frollR.o frolladaptive.o fsort.o fwrite.o fwriteR.o gsumm.o idatetime.o ijoin.o init.o inrange.o nafill.o negate.o nqrecreateindices.o openmp-utils.o programming.o quickselect.o rbindlist.o reorder.o shift.o snprintf.o subset.o transpose.o types.o uniqlist.o utils.o vecseq.o wrappers.o -fopenmp -L/usr/lib/x86_64-linux-gnu -lz -L/usr/lib/R/lib -lR\n",
      "PKG_CFLAGS = -fopenmp\n",
      "PKG_LIBS = -fopenmp -L/usr/lib/x86_64-linux-gnu -lz\n",
      "if [ \"data.table.so\" != \"data_table.so\" ]; then mv data.table.so data_table.so; fi\n",
      "if [ \"\" != \"Windows_NT\" ] && [ `uname -s` = 'Darwin' ]; then install_name_tool -id data_table.so data_table.so; fi\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "installing to /home/jupyter/packages/00LOCK-data.table/00new/data.table/libs\n",
      "** R\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** checking absolute paths in shared objects and dynamic libraries\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (data.table)\n",
      "* installing *source* package ‘qqman’ ...\n",
      "** package ‘qqman’ successfully unpacked and MD5 sums checked\n",
      "** using staged installation\n",
      "** R\n",
      "** data\n",
      "*** moving datasets to lazyload DB\n",
      "** inst\n",
      "** byte-compile and prepare package for lazy loading\n",
      "** help\n",
      "*** installing help indices\n",
      "** building package indices\n",
      "** installing vignettes\n",
      "** testing if installed package can be loaded from temporary location\n",
      "** testing if installed package can be loaded from final location\n",
      "** testing if installed package keeps a record of temporary installation path\n",
      "* DONE (qqman)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──\n",
      "✔ dplyr     1.1.4     ✔ readr     2.1.5\n",
      "✔ forcats   1.0.0     ✔ stringr   1.5.1\n",
      "✔ ggplot2   3.5.1     ✔ tibble    3.2.1\n",
      "✔ lubridate 1.9.4     ✔ tidyr     1.3.1\n",
      "✔ purrr     1.0.4     \n",
      "── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "✖ dplyr::filter() masks stats::filter()\n",
      "✖ dplyr::lag()    masks stats::lag()\n",
      "ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "Installing package into ‘/home/jupyter/packages’\n",
       "(as ‘lib’ is unspecified)\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/tidyverse_2.0.0.tar.gz'\n",
       "Content type 'application/x-gzip' length 704618 bytes (688 KB)\n",
       "==================================================\n",
       "downloaded 688 KB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/RtmpEei0DZ/downloaded_packages’\n",
       "Installing package into ‘/home/jupyter/packages’\n",
       "(as ‘lib’ is unspecified)\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/data.table_1.17.4.tar.gz'\n",
       "Content type 'application/x-gzip' length 5839682 bytes (5.6 MB)\n",
       "==================================================\n",
       "downloaded 5.6 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/RtmpEei0DZ/downloaded_packages’\n",
       "Installing package into ‘/home/jupyter/packages’\n",
       "(as ‘lib’ is unspecified)\n",
       "trying URL 'https://cloud.r-project.org/src/contrib/qqman_0.1.9.tar.gz'\n",
       "Content type 'application/x-gzip' length 1337980 bytes (1.3 MB)\n",
       "==================================================\n",
       "downloaded 1.3 MB\n",
       "\n",
       "\n",
       "The downloaded source packages are in\n",
       "\t‘/tmp/RtmpEei0DZ/downloaded_packages’\n",
       "data.table 1.17.4 using 4 threads (see ?getDTthreads).  Latest news: r-datatable.com\n",
       "\n",
       "Attaching package: ‘data.table’\n",
       "\n",
       "The following objects are masked from ‘package:lubridate’:\n",
       "\n",
       "    hour, isoweek, mday, minute, month, quarter, second, wday, week,\n",
       "    yday, year\n",
       "\n",
       "The following objects are masked from ‘package:dplyr’:\n",
       "\n",
       "    between, first, last\n",
       "\n",
       "The following object is masked from ‘package:purrr’:\n",
       "\n",
       "    transpose\n",
       "\n",
       "\n",
       "For example usage please run: vignette('qqman')\n",
       "\n",
       "Citation appreciated but not required:\n",
       "Turner, (2018). qqman: an R package for visualizing GWAS results using Q-Q and manhattan plots. Journal of Open Source Software, 3(25), 731, https://doi.org/10.21105/joss.00731.\n",
       "\n",
       "In addition: Warning message:\n",
       "In (function (package, help, pos = 2, lib.loc = NULL, character.only = FALSE,  :\n",
       "  library ‘/usr/lib/R/site-library’ contains no packages\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%%R\n",
    "install.packages(\"tidyverse\")\n",
    "install.packages(\"data.table\")\n",
    "install.packages(\"qqman\")\n",
    "\n",
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(qqman)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Remove related individuals"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAC 10\n",
      "AFR 69\n",
      "AJ 139\n",
      "AMR 58\n",
      "CAH 22\n",
      "CAS 74\n",
      "EAS 227\n",
      "EUR 1407\n",
      "MDE 22\n",
      "SAS 46\n"
     ]
    }
   ],
   "source": [
    "labels = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "\n",
    "for label in labels:\n",
    "    df = pd.read_csv(f'{WORK_DIR}/{label}/{label}_release9.related')\n",
    "    related_list = ['first_deg', 'second_deg']\n",
    "    related = df[df['REL'].isin(related_list)]\n",
    "    \n",
    "    # Keep one of the samples that flag as related\n",
    "    related = related.drop_duplicates(subset = 'IID1', keep = 'first')\n",
    "    \n",
    "    # Print out the numbers of related individuals\n",
    "    print(label, len(related))\n",
    "    \n",
    "    # Save out files\n",
    "    related['IID1'].to_csv(f'{WORK_DIR}/{label}/{label}_related_ids.samples', sep = '\\t', header = False, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    # Remove the \"_s1\" at the end of the samples labeling to match with PLINK files\n",
    "    \n",
    "    sed 's/_s1//' \"${label}/${label}_related_ids.samples\" > tmp\n",
    "    mv tmp \"${label}/${label}_related_ids.samples\"\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink2 \\\n",
    "        --pfile \"${label}/${label}_release9\" \\\n",
    "        --remove \"${label}/${label}_related_ids.samples\" \\\n",
    "        --make-bed \\\n",
    "        --out \"${label}/${label}_release9_remove_related\"\n",
    "\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# File preparation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## PLINK file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Recode the PHENOTYPE AND SEX as 1 and 2\n",
    "master = pd.read_csv(f\"{WORK_DIR}/master_key_release9_final.csv\")\n",
    "\n",
    "master.loc[(master[\"baseline_GP2_phenotype\"] != \"PD\") & (master[\"baseline_GP2_phenotype\"] != \"Control\"), \"baseline_GP2_phenotype\"] = -9\n",
    "master.loc[master[\"baseline_GP2_phenotype\"] == \"Control\", \"baseline_GP2_phenotype\"] = 1\n",
    "master.loc[master[\"baseline_GP2_phenotype\"] == \"PD\", \"baseline_GP2_phenotype\"] = 2\n",
    "\n",
    "master.loc[master[\"biological_sex_for_qc\"] == \"Male\", \"biological_sex_for_qc\"] = 1\n",
    "master.loc[master[\"biological_sex_for_qc\"] == \"Female\", \"biological_sex_for_qc\"] = 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename the columns\n",
    "master.rename(columns={\"baseline_GP2_phenotype\":\"PHENO\"}, inplace = True)\n",
    "master.rename(columns={\"biological_sex_for_qc\":\"SEX\"}, inplace = True)\n",
    "master.rename(columns={\"age_at_sample_collection\" : \"AGE\", \"GP2ID\":\"IID\"}, inplace = True)\n",
    "master.rename(columns={\"family_history_for_qc\":\"FHX\"}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save it to update PLINK file\n",
    "\n",
    "sex = master[[\"IID\", \"SEX\"]]\n",
    "pheno = master[[\"IID\", \"PHENO\"]]\n",
    "\n",
    "sex.insert(0, \"FID\", 0)\n",
    "pheno.insert(0, \"FID\", 0)\n",
    "\n",
    "sex.to_csv(f\"{WORK_DIR}/sex_update.txt\", sep = \"\\t\", header = False, index = False)\n",
    "pheno.to_csv(f\"{WORK_DIR}/pheno_update.txt\", sep = \"\\t\", header = False, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "cd $WORK_DIR\n",
    "\n",
    "labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'FIN' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${labels[@]}\"\n",
    "do\n",
    "    /home/jupyter/plink1.9 \\\n",
    "        --bfile \"${label}/${label}_release9_remove_related\" \\\n",
    "        --update-sex sex_update.txt \\\n",
    "        --pheno pheno_update.txt \\\n",
    "        --make-bed \\\n",
    "        --out \"${label}/${label}_release9_remove_related_updated\"\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Covariate file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "\n",
    "for label in labels:\n",
    "\n",
    "    # Keep separate covariate files per ancestry\n",
    "    master_red = master[(master[\"nba_label\"] == label) & (master[\"nba\"] == 1)]\n",
    "    master_red_base = master_red[[\"IID\", \"PHENO\", \"SEX\", \"AGE\", \"FHX\"]]\n",
    "    master_red_base.to_csv(f\"{WORK_DIR}/{label}/{label}_other_cov.txt\", sep = \"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "labels = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "\n",
    "for label in labels:\n",
    "    \n",
    "    fam = pd.read_csv(f'{WORK_DIR}/{label}/{label}_release9_remove_related_updated.fam', delim_whitespace = True, names = ['FID', 'IID', 'PATID', 'FATID', 'SEX', 'PHENO'])\n",
    "    fam = fam[['FID', 'IID', 'SEX', 'PHENO']]\n",
    "    \n",
    "    new_pcs = pd.read_csv(f'{WORK_DIR}/{label}/{label}_release9.eigenvec', sep = '\\t')\n",
    "    new_pcs.columns =['FID', 'IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10']\n",
    "    new_pcs = new_pcs[['FID', 'IID', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10']]\n",
    "    \n",
    "    # Merge it with PCs information to get full covariate file\n",
    "    merge1 = pd.merge(fam, new_pcs, on = ['FID', 'IID'])\n",
    "    merge1 = merge1[~merge1['PHENO'].isna()]\n",
    "    merge1['PHENO'] = merge1['PHENO'].astype(int)\n",
    "    \n",
    "    age = master[[\"IID\", \"AGE\"]]\n",
    "    merge2 = pd.merge(merge1, age, how = \"left\")\n",
    "    \n",
    "    merge2.insert(2, \"PAT\", 0)\n",
    "    merge2.insert(2, \"MAT\", 0)\n",
    "    merge2 = merge2[['FID', 'IID','SEX', 'PHENO', 'AGE', 'PC1', 'PC2', 'PC3', 'PC4', 'PC5', 'PC6', 'PC7', 'PC8', 'PC9', 'PC10']]\n",
    "    \n",
    "    merge2.to_csv(f'{WORK_DIR}/{label}/{label}_covar.txt', sep = '\\t',  index = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Demographics Documentation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Phenotype per ancestry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nCon</th>\n",
       "      <th>pCon</th>\n",
       "      <th>nCase</th>\n",
       "      <th>pCase</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>827</td>\n",
       "      <td>70.99</td>\n",
       "      <td>338</td>\n",
       "      <td>29.01</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1667</td>\n",
       "      <td>62.91</td>\n",
       "      <td>983</td>\n",
       "      <td>37.09</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>824</td>\n",
       "      <td>32.53</td>\n",
       "      <td>1709</td>\n",
       "      <td>67.47</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1428</td>\n",
       "      <td>41.60</td>\n",
       "      <td>2005</td>\n",
       "      <td>58.40</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>310</td>\n",
       "      <td>32.49</td>\n",
       "      <td>644</td>\n",
       "      <td>67.51</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>329</td>\n",
       "      <td>33.23</td>\n",
       "      <td>661</td>\n",
       "      <td>66.77</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2379</td>\n",
       "      <td>42.70</td>\n",
       "      <td>3192</td>\n",
       "      <td>57.30</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>5492</td>\n",
       "      <td>26.37</td>\n",
       "      <td>15332</td>\n",
       "      <td>73.63</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>197</td>\n",
       "      <td>29.06</td>\n",
       "      <td>481</td>\n",
       "      <td>70.94</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>199</td>\n",
       "      <td>35.99</td>\n",
       "      <td>354</td>\n",
       "      <td>64.01</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nCon   pCon  nCase  pCase label\n",
       "0   827  70.99    338  29.01   AAC\n",
       "0  1667  62.91    983  37.09   AFR\n",
       "0   824  32.53   1709  67.47    AJ\n",
       "0  1428  41.60   2005  58.40   AMR\n",
       "0   310  32.49    644  67.51   CAH\n",
       "0   329  33.23    661  66.77   CAS\n",
       "0  2379  42.70   3192  57.30   EAS\n",
       "0  5492  26.37  15332  73.63   EUR\n",
       "0   197  29.06    481  70.94   MDE\n",
       "0   199  35.99    354  64.01   SAS"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pheno_df = pd.DataFrame()\n",
    "lables = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "for label in lables:\n",
    "    \n",
    "    fam      = pd.read_csv(f'{WORK_DIR}/{label}/{label}_release9_remove_related_updated.fam', delim_whitespace = True, names = [\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENO\"])\n",
    "    fam_filt = fam[fam[\"PHENO\"] != -9]\n",
    "    \n",
    "    # Number of controls and cases\n",
    "    nCon    = fam_filt[fam_filt[\"PHENO\"] == 1].shape[0]\n",
    "    nCase   = fam_filt[fam_filt[\"PHENO\"] == 2].shape[0]\n",
    "    \n",
    "    # Percentage of controls and cases\n",
    "    pCon    = round( (nCon / (nCon + nCase)) * 100, 2)\n",
    "    pCase     = round( (nCase / (nCon + nCase)) * 100, 2)\n",
    "    \n",
    "    tmp      = pd.DataFrame([{\"nCon\": nCon, \"pCon\": pCon, \"nCase\": nCase, \"pCase\": pCase, \"label\" : label}])\n",
    "    pheno_df   = pd.concat([pheno_df, tmp])\n",
    "    \n",
    "pheno_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Sex per ancestry"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nMale</th>\n",
       "      <th>pMale</th>\n",
       "      <th>nFem</th>\n",
       "      <th>pFem</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>481</td>\n",
       "      <td>41.29</td>\n",
       "      <td>684</td>\n",
       "      <td>58.71</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1470</td>\n",
       "      <td>55.47</td>\n",
       "      <td>1180</td>\n",
       "      <td>44.53</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1553</td>\n",
       "      <td>61.31</td>\n",
       "      <td>980</td>\n",
       "      <td>38.69</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1631</td>\n",
       "      <td>47.51</td>\n",
       "      <td>1802</td>\n",
       "      <td>52.49</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>500</td>\n",
       "      <td>52.41</td>\n",
       "      <td>454</td>\n",
       "      <td>47.59</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>452</td>\n",
       "      <td>45.66</td>\n",
       "      <td>538</td>\n",
       "      <td>54.34</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3454</td>\n",
       "      <td>62.00</td>\n",
       "      <td>2117</td>\n",
       "      <td>38.00</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>12627</td>\n",
       "      <td>60.64</td>\n",
       "      <td>8197</td>\n",
       "      <td>39.36</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>417</td>\n",
       "      <td>61.50</td>\n",
       "      <td>261</td>\n",
       "      <td>38.50</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>358</td>\n",
       "      <td>64.74</td>\n",
       "      <td>195</td>\n",
       "      <td>35.26</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nMale  pMale  nFem   pFem label\n",
       "0    481  41.29   684  58.71   AAC\n",
       "0   1470  55.47  1180  44.53   AFR\n",
       "0   1553  61.31   980  38.69    AJ\n",
       "0   1631  47.51  1802  52.49   AMR\n",
       "0    500  52.41   454  47.59   CAH\n",
       "0    452  45.66   538  54.34   CAS\n",
       "0   3454  62.00  2117  38.00   EAS\n",
       "0  12627  60.64  8197  39.36   EUR\n",
       "0    417  61.50   261  38.50   MDE\n",
       "0    358  64.74   195  35.26   SAS"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sex_df = pd.DataFrame()\n",
    "lables = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "for label in lables:\n",
    "    \n",
    "    fam      = pd.read_csv(f'{WORK_DIR}/{label}/{label}_release9_remove_related_updated.fam', delim_whitespace = True, names = [\"FID\", \"IID\", \"PAT\", \"MAT\", \"SEX\", \"PHENO\"])\n",
    "    fam_filt = fam[fam[\"PHENO\"] != -9]\n",
    "    \n",
    "    # Number of male and female\n",
    "    nMale    = fam_filt[fam_filt[\"SEX\"] == 1].shape[0]\n",
    "    nFem     = fam_filt[fam_filt[\"SEX\"] == 2].shape[0]\n",
    "    \n",
    "    # Percentage of male and female\n",
    "    pMale    = round( (nMale / (nMale + nFem)) * 100, 2)\n",
    "    pFem     = round( (nFem / (nMale + nFem)) * 100, 2)\n",
    "    \n",
    "    tmp      = pd.DataFrame([{\"nMale\": nMale, \"pMale\": pMale, \"nFem\": nFem, \"pFem\": pFem, \"label\" : label}])\n",
    "    sex_df   = pd.concat([sex_df, tmp])\n",
    "    \n",
    "sex_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Age "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nComplete</th>\n",
       "      <th>nMissing</th>\n",
       "      <th>mean_age</th>\n",
       "      <th>std</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>775</td>\n",
       "      <td>52</td>\n",
       "      <td>64.94</td>\n",
       "      <td>11.46</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>949</td>\n",
       "      <td>718</td>\n",
       "      <td>62.50</td>\n",
       "      <td>16.00</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>627</td>\n",
       "      <td>197</td>\n",
       "      <td>63.11</td>\n",
       "      <td>11.47</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1374</td>\n",
       "      <td>54</td>\n",
       "      <td>59.83</td>\n",
       "      <td>8.45</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>299</td>\n",
       "      <td>11</td>\n",
       "      <td>47.20</td>\n",
       "      <td>19.33</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>239</td>\n",
       "      <td>90</td>\n",
       "      <td>54.86</td>\n",
       "      <td>6.13</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1307</td>\n",
       "      <td>1072</td>\n",
       "      <td>62.41</td>\n",
       "      <td>11.13</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3179</td>\n",
       "      <td>2313</td>\n",
       "      <td>64.72</td>\n",
       "      <td>12.51</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>193</td>\n",
       "      <td>4</td>\n",
       "      <td>55.58</td>\n",
       "      <td>8.84</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>123</td>\n",
       "      <td>76</td>\n",
       "      <td>54.15</td>\n",
       "      <td>17.43</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nComplete  nMissing  mean_age    std label\n",
       "0        775        52     64.94  11.46   AAC\n",
       "0        949       718     62.50  16.00   AFR\n",
       "0        627       197     63.11  11.47    AJ\n",
       "0       1374        54     59.83   8.45   AMR\n",
       "0        299        11     47.20  19.33   CAH\n",
       "0        239        90     54.86   6.13   CAS\n",
       "0       1307      1072     62.41  11.13   EAS\n",
       "0       3179      2313     64.72  12.51   EUR\n",
       "0        193         4     55.58   8.84   MDE\n",
       "0        123        76     54.15  17.43   SAS"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Control Age\n",
    "age_df = pd.DataFrame()\n",
    "lables = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "for label in lables:\n",
    "    \n",
    "    covar          = pd.read_csv(f'{WORK_DIR}/{label}/{label}_covar.txt', sep = '\\t')\n",
    "    covar_con      = covar[covar[\"PHENO\"] == 1]\n",
    "    covar_con_filt = covar_con[~covar_con['AGE'].isna()]\n",
    "    \n",
    "    # Calculate number of individuals with missing age \n",
    "    nMissing  = covar_con[covar_con['AGE'].isna()].shape[0]\n",
    "    nComplete = covar_con[~covar_con['AGE'].isna()].shape[0]\n",
    "    \n",
    "    mean_age  = round(covar_con_filt['AGE'].mean(), 2)\n",
    "    stdev     = round(covar_con_filt['AGE'].std(), 2) # Standard deviation\n",
    "    tmp       = pd.DataFrame([{\"nComplete\": nComplete, \"nMissing\": nMissing, \"mean_age\":mean_age, \"std\":stdev, \"label\" : label}])\n",
    "    age_df    = pd.concat([age_df, tmp])\n",
    "    \n",
    "age_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nComplete</th>\n",
       "      <th>nMissing</th>\n",
       "      <th>mean_age</th>\n",
       "      <th>std</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>287</td>\n",
       "      <td>51</td>\n",
       "      <td>65.90</td>\n",
       "      <td>11.55</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>270</td>\n",
       "      <td>713</td>\n",
       "      <td>63.97</td>\n",
       "      <td>11.82</td>\n",
       "      <td>AFR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1617</td>\n",
       "      <td>92</td>\n",
       "      <td>70.26</td>\n",
       "      <td>9.75</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1916</td>\n",
       "      <td>89</td>\n",
       "      <td>62.83</td>\n",
       "      <td>12.25</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>556</td>\n",
       "      <td>88</td>\n",
       "      <td>61.51</td>\n",
       "      <td>12.44</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>528</td>\n",
       "      <td>133</td>\n",
       "      <td>61.13</td>\n",
       "      <td>10.87</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2612</td>\n",
       "      <td>580</td>\n",
       "      <td>64.26</td>\n",
       "      <td>11.54</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11906</td>\n",
       "      <td>3426</td>\n",
       "      <td>67.27</td>\n",
       "      <td>10.76</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>333</td>\n",
       "      <td>148</td>\n",
       "      <td>64.07</td>\n",
       "      <td>12.23</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>316</td>\n",
       "      <td>38</td>\n",
       "      <td>60.57</td>\n",
       "      <td>12.46</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nComplete  nMissing  mean_age    std label\n",
       "0        287        51     65.90  11.55   AAC\n",
       "0        270       713     63.97  11.82   AFR\n",
       "0       1617        92     70.26   9.75    AJ\n",
       "0       1916        89     62.83  12.25   AMR\n",
       "0        556        88     61.51  12.44   CAH\n",
       "0        528       133     61.13  10.87   CAS\n",
       "0       2612       580     64.26  11.54   EAS\n",
       "0      11906      3426     67.27  10.76   EUR\n",
       "0        333       148     64.07  12.23   MDE\n",
       "0        316        38     60.57  12.46   SAS"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# PD Age\n",
    "age_df = pd.DataFrame()\n",
    "lables = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "for label in lables:\n",
    "    \n",
    "    covar           = pd.read_csv(f'{WORK_DIR}/{label}/{label}_covar.txt', sep = '\\t')\n",
    "    covar_case      = covar[covar[\"PHENO\"] == 2]\n",
    "    covar_case_filt = covar_case[~covar_case['AGE'].isna()]\n",
    "    \n",
    "    # Calculate number of individuals with missing age\n",
    "    nMissing  = covar_case[covar_case['AGE'].isna()].shape[0]\n",
    "    nComplete = covar_case[~covar_case['AGE'].isna()].shape[0]\n",
    "    \n",
    "    mean_age  = round(covar_case_filt['AGE'].mean(), 2)\n",
    "    stdev     = round(covar_case_filt['AGE'].std(), 2) # Standard deviation\n",
    "    tmp       = pd.DataFrame([{\"nComplete\": nComplete, \"nMissing\": nMissing, \"mean_age\":mean_age, \"std\":stdev, \"label\" : label}])\n",
    "    age_df    = pd.concat([age_df, tmp])\n",
    "    \n",
    "age_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare differences between p.A419V PD carriers and PD non-carriers "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "lables = ['AAC', 'AFR', 'AJ', 'AMR', 'CAH', 'CAS', 'EAS', 'EUR', 'MDE', 'SAS']\n",
    "for label in lables:\n",
    "\n",
    "    master_red = master[(master[\"nba_label\"] == label) & (master[\"nba\"] == 1)]\n",
    "    master_red_base = master_red[[\"IID\", \"PHENO\", \"SEX\", \"AGE\", \"FHX\"]]\n",
    "    master_red_base.to_csv(f\"{WORK_DIR}/{label}/{label}_other_cov.txt\", sep = \"\\t\", header = True, index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "lables=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "# Extract p.A419V\n",
    "for label in \"${lables[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_updated \\\n",
    "    --chr 12 \\\n",
    "    --from-bp 40252984 \\\n",
    "    --to-bp 40252984 \\\n",
    "    --missing \\\n",
    "    --make-bed \\\n",
    "    --out ${label}/${label}_release9_remove_related_a419v\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "lables=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${lables[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_a419v \\\n",
    "    --recode A \\\n",
    "    --out ${label}/${label}_release9_remove_related_a419v\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Family History"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      anc ntotal  nFHX nWithoutFHX     p_val FHx_c FHx_nc total_c total_nc\n",
      "   <char>  <int> <int>       <int>     <num> <int>  <int>   <int>    <int>\n",
      "1:    CAH    508   102         406 0.1814038     2    100       4      504\n",
      "2:    EUR  11630  2959        8671 0.3559985     3   2953      11    11610\n",
      "3:    EAS   2711   664        2047 0.2906378     8    656      53     2657\n",
      "4:    CAS    516    59         457 1.0000000     1     58      11      504\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "labels <- c(\"CAH\", \"EUR\", \"EAS\", \"CAS\")\n",
    "\n",
    "df <- data.table()\n",
    "\n",
    "for (label in labels){\n",
    "    \n",
    "    raw <- read.delim(paste0(\"/home/jupyter/A419V_release9/\", label, \"/\", label, \"_release9_remove_related_a419v.raw\"), sep = \" \")\n",
    "    cov <- read.delim(paste0(\"/home/jupyter/A419V_release9/\", label, \"/\", label, \"_other_cov.txt\"), sep = \"\\t\")\n",
    "\n",
    "    merged <- merge(raw, cov, by = \"IID\")\n",
    "\n",
    "    merge_filtered <- merged %>% \n",
    "                      filter(PHENOTYPE == 2) %>%\n",
    "                      subset(FHX %in% c(\"No\", \"Yes\"))\n",
    "\n",
    "    table_fhx <- table(merge_filtered$exm994472_A, merge_filtered$FHX)\n",
    "\n",
    "    p          <- fisher.test(table_fhx)$p\n",
    "    total      <- nrow(merge_filtered)\n",
    "    withFHx    <- merge_filtered %>% filter(FHX == \"Yes\") %>% nrow()\n",
    "    withoutFHx <- merge_filtered %>% filter(FHX == \"No\") %>% nrow()\n",
    "    \n",
    "    withFHx_carrier <- merge_filtered %>% filter(FHX == \"Yes\" & exm994472_A != 0) %>% nrow()\n",
    "    withFHx_noncarrier <- merge_filtered %>% filter(FHX == \"Yes\" & exm994472_A == 0) %>% nrow()\n",
    "\n",
    "    total_carrier <- merge_filtered %>% filter(exm994472_A != 0) %>% nrow()\n",
    "    total_noncarrier <- merge_filtered %>% filter(exm994472_A == 0) %>% nrow()\n",
    "    \n",
    "    tmp <- data.table(anc = label, ntotal = total, nFHX = withFHx, nWithoutFHX = withoutFHx, p_val = p, \n",
    "                      FHx_c = withFHx_carrier, FHx_nc = withFHx_noncarrier,\n",
    "                      total_c = total_carrier, total_nc = total_noncarrier)\n",
    "    df  <- rbind(df, tmp)\n",
    "\n",
    "}\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "      anc ntotal nMale nFemale       p_val male_c male_nc total_c total_nc\n",
      "   <char>  <int> <int>   <int>       <num>  <int>   <int>   <int>    <int>\n",
      "1:    CAH    644   369     275 0.051800339      4     365      14      630\n",
      "2:    EUR  15332  9549    5783 0.040907116     15    9528      33    15288\n",
      "3:    EAS   3192  1767    1425 0.087092101     37    1729      81     3109\n",
      "4:    CAS    661   301     360 0.008410676      1     299      13      647\n"
     ]
    }
   ],
   "source": [
    "%%R\n",
    "\n",
    "labels <- c(\"CAH\", \"EUR\", \"EAS\", \"CAS\")\n",
    "\n",
    "df <- data.table()\n",
    "\n",
    "for (label in labels){\n",
    "    \n",
    "    raw <- read.delim(paste0(\"/home/jupyter/A419V_release9/\", label, \"/\", label, \"_release9_remove_related_a419v.raw\"), sep = \" \")\n",
    "    cov <- read.delim(paste0(\"/home/jupyter/A419V_release9/\", label, \"/\", label, \"_other_cov.txt\"), sep = \"\\t\")\n",
    "\n",
    "    merged <- merge(raw, cov, by = \"IID\")\n",
    "\n",
    "    merge_filtered <- merged %>% \n",
    "                      filter(PHENOTYPE == 2) %>%\n",
    "                      subset(SEX.y %in% c(1, 2))\n",
    "\n",
    "    table_sex <- table(merge_filtered$exm994472_A, merge_filtered$SEX.y)\n",
    "\n",
    "    p          <- fisher.test(table_sex)$p\n",
    "    total      <- nrow(merge_filtered)\n",
    "    male       <- merge_filtered %>% filter(SEX.y == 1) %>% nrow()\n",
    "    female     <- merge_filtered %>% filter(SEX.y == 2) %>% nrow()\n",
    "    \n",
    "    male_carrier       <- merge_filtered %>% filter(SEX.y == 1 & exm994472_A != 0) %>% nrow()\n",
    "    male_noncarrier    <- merge_filtered %>% filter(SEX.y == 1 & exm994472_A == 0) %>% nrow()\n",
    "    \n",
    "    total_carrier <- merge_filtered %>% filter(exm994472_A != 0) %>% nrow()\n",
    "    total_noncarrier <- merge_filtered %>% filter(exm994472_A == 0) %>% nrow()\n",
    "    \n",
    "    tmp <- data.table(anc = label, ntotal = total, nMale = male, nFemale = female, p_val = p, \n",
    "                      male_c = male_carrier, male_nc = male_noncarrier, \n",
    "                      total_c = total_carrier, total_nc = total_noncarrier)\n",
    "    df  <- rbind(df, tmp)\n",
    "\n",
    "}\n",
    "\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HWE check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "lables=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${lables[@]}\"\n",
    "do\n",
    "\n",
    "    /home/jupyter/plink1.9 \\\n",
    "    --bfile ${label}/${label}_release9_remove_related_a419v \\\n",
    "    --hardy \\\n",
    "    --out ${label}/${label}_release9_remove_related_a419v_hwe\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AAC\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G             0/0/1203        0        0            1\n",
      "  12   Seq_rs34594498      AFF    A    G              0/0/338        0        0            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/0/824        0        0            1\n",
      "  12        exm994472      ALL    A    G             0/0/1207        0        0            1\n",
      "  12        exm994472      AFF    A    G              0/0/338        0        0            1\n",
      "  12        exm994472    UNAFF    A    G              0/0/827        0        0            1\n",
      "AFR\n",
      "AJ\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G             0/1/3076 0.000325 0.0003249            1\n",
      "  12   Seq_rs34594498      AFF    A    G             0/1/1705 0.0005862 0.000586            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/0/824        0        0            1\n",
      "  12        exm994472      ALL    A    G             0/1/3078 0.0003248 0.0003247            1\n",
      "  12        exm994472      AFF    A    G             0/1/1707 0.0005855 0.0005853            1\n",
      "  12        exm994472    UNAFF    A    G              0/0/824        0        0            1\n",
      "AMR\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "cat: AFR/AFR_release9_remove_related_a419v_hwe.hwe: No such file or directory\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G             0/1/3492 0.0002863 0.0002862            1\n",
      "  12   Seq_rs34594498      AFF    A    G             0/0/2005        0        0            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G             0/0/1428        0        0            1\n",
      "  12        exm994472      ALL    A    G             0/1/3491 0.0002864 0.0002863            1\n",
      "  12        exm994472      AFF    A    G             0/0/2004        0        0            1\n",
      "  12        exm994472    UNAFF    A    G             0/0/1428        0        0            1\n",
      "CAH\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G             0/13/968  0.01325  0.01316            1\n",
      "  12   Seq_rs34594498      AFF    A    G             0/13/630  0.02022  0.02001            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/0/310        0        0            1\n",
      "  12        exm994472      ALL    A    G             0/14/968  0.01426  0.01415            1\n",
      "  12        exm994472      AFF    A    G             0/14/630  0.02174   0.0215            1\n",
      "  12        exm994472    UNAFF    A    G              0/0/310        0        0            1\n",
      "CAS\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G             0/17/988  0.01692  0.01677            1\n",
      "  12   Seq_rs34594498      AFF    A    G             0/13/648  0.01967  0.01947            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/4/324   0.0122  0.01212            1\n",
      "  12        exm994472      ALL    A    G             0/17/988  0.01692  0.01677            1\n",
      "  12        exm994472      AFF    A    G             0/13/647   0.0197   0.0195            1\n",
      "  12        exm994472    UNAFF    A    G              0/4/325  0.01216  0.01208            1\n",
      "EAS\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G            2/95/5540  0.01685  0.01741      0.06844\n",
      "  12   Seq_rs34594498      AFF    A    G            1/80/3109  0.02508  0.02537       0.4098\n",
      "  12   Seq_rs34594498    UNAFF    A    G            1/14/2357 0.005902 0.006723      0.02506\n",
      "  12        exm994472      ALL    A    G            2/95/5538  0.01686  0.01741      0.06849\n",
      "  12        exm994472      AFF    A    G            1/80/3109  0.02508  0.02537       0.4098\n",
      "  12        exm994472    UNAFF    A    G            1/14/2355 0.005907 0.006728      0.02508\n",
      "EUR\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G           2/36/34630 0.001038 0.001153    5.669e-05\n",
      "  12   Seq_rs34594498      AFF    A    G           2/30/15288 0.001958 0.002217    0.0001468\n",
      "  12   Seq_rs34594498    UNAFF    A    G             0/2/5481 0.0003648 0.0003647            1\n",
      "  12        exm994472      ALL    A    G           2/39/34638 0.001125 0.001239    7.642e-05\n",
      "  12        exm994472      AFF    A    G           2/31/15288 0.002023 0.002282    0.0001656\n",
      "  12        exm994472    UNAFF    A    G             0/2/5485 0.0003645 0.0003644            1\n",
      "MDE\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G              0/0/711        0        0            1\n",
      "  12   Seq_rs34594498      AFF    A    G              0/0/481        0        0            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/0/196        0        0            1\n",
      "  12        exm994472      ALL    A    G              0/0/711        0        0            1\n",
      "  12        exm994472      AFF    A    G              0/0/481        0        0            1\n",
      "  12        exm994472    UNAFF    A    G              0/0/196        0        0            1\n",
      "SAS\n",
      " CHR              SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P \n",
      "  12   Seq_rs34594498      ALL    A    G              0/0/580        0        0            1\n",
      "  12   Seq_rs34594498      AFF    A    G              0/0/354        0        0            1\n",
      "  12   Seq_rs34594498    UNAFF    A    G              0/0/199        0        0            1\n",
      "  12        exm994472      ALL    A    G              0/0/579        0        0            1\n",
      "  12        exm994472      AFF    A    G              0/0/353        0        0            1\n",
      "  12        exm994472    UNAFF    A    G              0/0/199        0        0            1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "WORK_DIR=/home/jupyter/A419V_release9\n",
    "cd $WORK_DIR\n",
    "\n",
    "ancestry_labels=('AAC' 'AFR' 'AJ' 'AMR' 'CAH' 'CAS' 'EAS' 'EUR' 'MDE' 'SAS')\n",
    "\n",
    "for label in \"${ancestry_labels[@]}\";\n",
    "do\n",
    "    \n",
    "    echo ${label}\n",
    "    cat ${label}/${label}_release9_remove_related_a419v_hwe.hwe\n",
    "    \n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LRRK2 p.A419V allele frequency by sex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nCon</th>\n",
       "      <th>nCase</th>\n",
       "      <th>hom_ref</th>\n",
       "      <th>het</th>\n",
       "      <th>hom_alt</th>\n",
       "      <th>alt_freq</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>298</td>\n",
       "      <td>183</td>\n",
       "      <td>481</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>425</td>\n",
       "      <td>1128</td>\n",
       "      <td>1551</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000322</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>502</td>\n",
       "      <td>1129</td>\n",
       "      <td>1630</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>151</td>\n",
       "      <td>301</td>\n",
       "      <td>448</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0.003326</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>118</td>\n",
       "      <td>299</td>\n",
       "      <td>416</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>133</td>\n",
       "      <td>225</td>\n",
       "      <td>357</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1687</td>\n",
       "      <td>1767</td>\n",
       "      <td>3402</td>\n",
       "      <td>42</td>\n",
       "      <td>0</td>\n",
       "      <td>0.006098</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3078</td>\n",
       "      <td>9549</td>\n",
       "      <td>12602</td>\n",
       "      <td>16</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000634</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>131</td>\n",
       "      <td>369</td>\n",
       "      <td>496</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0.004000</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nCon  nCase  hom_ref  het  hom_alt  alt_freq label\n",
       "0   298    183      481    0        0  0.000000   AAC\n",
       "0   425   1128     1551    1        0  0.000322    AJ\n",
       "0   502   1129     1630    0        0  0.000000   AMR\n",
       "0   151    301      448    3        0  0.003326   CAS\n",
       "0   118    299      416    0        0  0.000000   MDE\n",
       "0   133    225      357    0        0  0.000000   SAS\n",
       "0  1687   1767     3402   42        0  0.006098   EAS\n",
       "0  3078   9549    12602   16        0  0.000634   EUR\n",
       "0   131    369      496    4        0  0.004000   CAH"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For Male\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "sex = 1\n",
    "\n",
    "geno_df = pd.DataFrame()\n",
    "ancestry_labels = ['AAC', 'AJ', 'AMR', 'CAS','MDE', 'SAS', 'EAS', 'EUR', 'CAH']\n",
    "\n",
    "for label in ancestry_labels:\n",
    "\n",
    "    raw = pd.read_csv(f\"{WORK_DIR}/{label}/{label}_release9_remove_related_a419v.raw\", delim_whitespace = True)\n",
    "    raw_male = raw[(raw[\"SEX\"] == sex) & (raw[\"PHENOTYPE\"] != -9)]\n",
    "    \n",
    "    nCon = raw_male[raw_male[\"PHENOTYPE\"] == 1].shape[0]\n",
    "    nCase = raw_male[raw_male[\"PHENOTYPE\"] == 2].shape[0]\n",
    "\n",
    "    hom_ref = raw_male[raw_male[\"exm994472_A\"] == 0].shape[0]\n",
    "    het     = raw_male[raw_male[\"exm994472_A\"] == 1].shape[0]\n",
    "    hom_alt = raw_male[raw_male[\"exm994472_A\"] == 2].shape[0]\n",
    "    \n",
    "    alt_freq = ( het + (hom_alt*2) ) / ( (hom_ref + het + hom_alt)*2 )\n",
    "    \n",
    "    tmp       = pd.DataFrame([{\"nCon\": nCon, \"nCase\": nCase, \"hom_ref\": hom_ref, \"het\": het, \"hom_alt\":hom_alt, \"alt_freq\": alt_freq, \"label\" : label}])\n",
    "    geno_df   = pd.concat([geno_df, tmp])\n",
    "\n",
    "geno_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nCon</th>\n",
       "      <th>nCase</th>\n",
       "      <th>hom_ref</th>\n",
       "      <th>het</th>\n",
       "      <th>hom_alt</th>\n",
       "      <th>alt_freq</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>529</td>\n",
       "      <td>155</td>\n",
       "      <td>684</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>AAC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>399</td>\n",
       "      <td>581</td>\n",
       "      <td>980</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>AJ</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>926</td>\n",
       "      <td>876</td>\n",
       "      <td>1802</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>AMR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>178</td>\n",
       "      <td>360</td>\n",
       "      <td>524</td>\n",
       "      <td>14</td>\n",
       "      <td>0</td>\n",
       "      <td>0.013011</td>\n",
       "      <td>CAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>79</td>\n",
       "      <td>182</td>\n",
       "      <td>261</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>MDE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66</td>\n",
       "      <td>129</td>\n",
       "      <td>195</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>692</td>\n",
       "      <td>1425</td>\n",
       "      <td>2062</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>0.013233</td>\n",
       "      <td>EAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2414</td>\n",
       "      <td>5783</td>\n",
       "      <td>8171</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>0.001282</td>\n",
       "      <td>EUR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>179</td>\n",
       "      <td>275</td>\n",
       "      <td>444</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0.011013</td>\n",
       "      <td>CAH</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   nCon  nCase  hom_ref  het  hom_alt  alt_freq label\n",
       "0   529    155      684    0        0  0.000000   AAC\n",
       "0   399    581      980    0        0  0.000000    AJ\n",
       "0   926    876     1802    0        0  0.000000   AMR\n",
       "0   178    360      524   14        0  0.013011   CAS\n",
       "0    79    182      261    0        0  0.000000   MDE\n",
       "0    66    129      195    0        0  0.000000   SAS\n",
       "0   692   1425     2062   52        2  0.013233   EAS\n",
       "0  2414   5783     8171   17        2  0.001282   EUR\n",
       "0   179    275      444   10        0  0.011013   CAH"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# For Female\n",
    "WORK_DIR='/home/jupyter/A419V_release9'\n",
    "sex = 2\n",
    "\n",
    "geno_df = pd.DataFrame()\n",
    "ancestry_labels = ['AAC', 'AJ', 'AMR', 'CAS','MDE', 'SAS', 'EAS', 'EUR', 'CAH']\n",
    "\n",
    "for label in ancestry_labels:\n",
    "\n",
    "    raw = pd.read_csv(f\"{WORK_DIR}/{label}/{label}_release9_remove_related_a419v.raw\", delim_whitespace = True)\n",
    "    raw_fem = raw[(raw[\"SEX\"] == sex) & (raw[\"PHENOTYPE\"] != -9)]\n",
    "    \n",
    "    nCon = raw_fem[raw_fem[\"PHENOTYPE\"] == 1].shape[0]\n",
    "    nCase = raw_fem[raw_fem[\"PHENOTYPE\"] == 2].shape[0]\n",
    "\n",
    "    hom_ref = raw_fem[raw_fem[\"exm994472_A\"] == 0].shape[0]\n",
    "    het     = raw_fem[raw_fem[\"exm994472_A\"] == 1].shape[0]\n",
    "    hom_alt = raw_fem[raw_fem[\"exm994472_A\"] == 2].shape[0]\n",
    "    \n",
    "    alt_freq = ( het + (hom_alt*2) ) / ( (hom_ref + het + hom_alt)*2 )\n",
    "    \n",
    "    tmp       = pd.DataFrame([{\"nCon\": nCon, \"nCase\": nCase, \"hom_ref\": hom_ref, \"het\": het, \"hom_alt\":hom_alt, \"alt_freq\": alt_freq, \"label\" : label}])\n",
    "    geno_df   = pd.concat([geno_df, tmp])\n",
    "\n",
    "geno_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
